diff --git a/megatron/core/dist_checkpointing/strategies/zarr.py b/megatron/core/dist_checkpointing/strategies/zarr.py
index 9b01403bf..443b46af3 100644
--- a/megatron/core/dist_checkpointing/strategies/zarr.py
+++ b/megatron/core/dist_checkpointing/strategies/zarr.py
@@ -175,11 +175,6 @@ def _create_zarr_array(sharded_tensor: ShardedTensor, checkpoint_dir: Path):
             compressor=None,
             fill_value=None,
             write_empty_chunks=True,
-            synchronizer=(
-                zarr.ProcessSynchronizer(str(checkpoint_dir / f'{sharded_tensor.key}.sync'))
-                if sharded_tensor.flattened_range is not None
-                else None
-            ),
         )
         logger.debug(f"Created a new Zarr array at {checkpoint_dir / sharded_tensor.key}")
     except zarr.errors.ContainsArrayError as e:
@@ -333,7 +328,7 @@ def load_zarr_based_sharded_metadata(
 
     sharded_state_dict = {}
     for subdir in checkpoint_dir.iterdir():
-        if not subdir.is_dir() or not (subdir / ".zarray").exists() or subdir.suffix == ".sync":
+        if not subdir.is_dir() or not (subdir / ".zarray").exists():
             continue
         key = subdir.name
         arr_shape, arr_dtype = get_shape_dtype_fn(str(subdir))

