diff --git a/megatron/core/dist_checkpointing/dict_utils.py b/megatron/core/dist_checkpointing/dict_utils.py
index eecf7674e..5e6fa506a 100644
--- a/megatron/core/dist_checkpointing/dict_utils.py
+++ b/megatron/core/dist_checkpointing/dict_utils.py
@@ -103,19 +103,11 @@ def diff(x1: Any, x2: Any, prefix: Tuple = ()) -> Tuple[list, list, list]:
     else:
         only_left = []
         only_right = []
-        mismatch_debug_data = [prefix, type(x1), type(x2)]
         if isinstance(x1, torch.Tensor) and isinstance(x2, torch.Tensor):
-            try:
-                if x1.device != x2.device:
-                    _is_mismatch = not torch.all(x1.cpu() == x2.cpu())
-                else:
-                    _is_mismatch = not torch.all(x1 == x2)
-                mismatch_debug_data.extend(
-                    [(x1 != x2).sum(), (x1 != x2).shape, (x1 != x2).nonzero().tolist()]
-                )
-            except (RuntimeError, TypeError, ValueError):
-                _is_mismatch = True
-                mismatch_debug_data.extend([x1.shape, x2.shape])
+            if x1.device != x2.device:
+                _is_mismatch = not torch.all(x1.cpu() == x2.cpu())
+            else:
+                _is_mismatch = not torch.all(x1 == x2)
         # TODO: change with concrete type that has both replica_id and data attrs
         elif hasattr(x1, "replica_id") and hasattr(x2, "replica_id"):
             assert type(x1) == type(x2)
@@ -130,7 +122,7 @@ def diff(x1: Any, x2: Any, prefix: Tuple = ()) -> Tuple[list, list, list]:
                 _is_mismatch = True
 
         if _is_mismatch:
-            mismatch.append(tuple(mismatch_debug_data))
+            mismatch.append((prefix, type(x1), type(x2)))
 
     return only_left, only_right, mismatch
 

