diff --git a/megatron/core/models/backends.py b/megatron/core/models/backends.py
index abda7c477..ac165dbbb 100644
--- a/megatron/core/models/backends.py
+++ b/megatron/core/models/backends.py
@@ -63,11 +63,6 @@ class BackendSpecProvider(Protocol):
         """Which module and submodules to use for grouped mlp"""
         ...
 
-    @abstractmethod
-    def activation_func(self) -> type:
-        """Which module to use for activation function"""
-        ...
-
 
 class LocalSpecProvider(BackendSpecProvider):
     """A protocol for providing Local submodules used in Spec building."""
@@ -115,7 +110,3 @@ class LocalSpecProvider(BackendSpecProvider):
             return SequentialMLP, MLPSubmodules(
                 linear_fc1=ColumnParallelLinear, linear_fc2=RowParallelLinear
             )
-
-    def activation_func(self) -> type:
-        """Which module to use for activation function"""
-        return None

