diff --git a/examples/post_training/modelopt/generate.py b/examples/post_training/modelopt/generate.py
index 4f490ba9f..5bc05927c 100644
--- a/examples/post_training/modelopt/generate.py
+++ b/examples/post_training/modelopt/generate.py
@@ -15,11 +15,11 @@ from megatron.post_training.arguments import add_modelopt_args
 from megatron.post_training.checkpointing import load_modelopt_checkpoint
 from megatron.post_training.generate import simple_generate
 from megatron.post_training.model_provider import model_provider
-from megatron.post_training.utils import report_current_memory_info, to_empty_if_meta
+from megatron.post_training.utils import report_current_memory_info
 from megatron.training import get_args, get_model, get_tokenizer, initialize_megatron
 from megatron.training.utils import print_rank_0, unwrap_model
 
-warnings.filterwarnings('once')
+warnings.filterwarnings('ignore')
 
 
 def add_generate_args(parser):
@@ -82,25 +82,6 @@ if __name__ == "__main__":
 
     args = get_args()
 
-    # Meta device initialization for ParallelLinear only works if using cpu initialization.
-    # Meta device initialization is used such that models can be materialized in low-precision
-    # directly when ModelOpt real quant is used. Otherwise, the model is first initialized
-    # as BF16 in memory which may result in OOM and defeat the purpose of real quant.
-    if args.init_model_with_meta_device:
-        args.use_cpu_initialization = True
-    else:
-        warnings.warn(
-            "--init-model-with-meta-device is not set. If you would like to resume the "
-            "model in low-bit directly (low-memory initialization and skipping 16-bit), "
-            "--init-model-with-meta-device must be set.",
-            UserWarning,
-        )
-
-    model = get_model(functools.partial(model_provider, parallel_output=True), wrap_with_ddp=False)
-    report_current_memory_info()
-
-    unwrapped_model = unwrap_model(model)[0]
-
     default_conversations = [
         {
             "role": "user",
@@ -119,7 +100,9 @@ if __name__ == "__main__":
         dataset = load_dataset(args.finetune_hf_dataset, split=args.finetune_data_split)
 
     tokenizer = get_tokenizer()._tokenizer
+    model = get_model(functools.partial(model_provider, parallel_output=True), wrap_with_ddp=False)
 
+    report_current_memory_info()
 
     if args.load is not None:
         load_modelopt_checkpoint(model, strict=not args.untie_embeddings_and_output_weights)

