diff --git a/examples/post_training/modelopt/mmlu.sh b/examples/post_training/modelopt/mmlu.sh
index 36918b89d..c62e7c435 100644
--- a/examples/post_training/modelopt/mmlu.sh
+++ b/examples/post_training/modelopt/mmlu.sh
@@ -11,7 +11,6 @@ MLM_DEFAULT_ARGS="--finetune --auto-detect-ckpt-format --export-te-mcore-model -
 ${LAUNCH_SCRIPT} ${SCRIPT_DIR}/mmlu.py \
     ${MODEL_ARGS} \
     --tensor-model-parallel-size ${TP} \
-    --expert-tensor-parallel-size ${ETP} \
     --expert-model-parallel-size ${EP} \
     --pipeline-model-parallel-size ${PP} \
     --tokenizer-model ${TOKENIZER_MODEL} \

