diff --git a/tests/unit_tests/transformer/test_spec_customization.py b/tests/unit_tests/transformer/test_spec_customization.py
index d6b2a2eb6..e2c5f47d6 100755
--- a/tests/unit_tests/transformer/test_spec_customization.py
+++ b/tests/unit_tests/transformer/test_spec_customization.py
@@ -16,7 +16,7 @@ from megatron.core.extensions.transformer_engine import (
 from megatron.core.fusions.fused_bias_dropout import get_bias_dropout_add
 from megatron.core.models.gpt.gpt_layer_specs import get_gpt_layer_local_spec
 from megatron.core.parallel_state import get_context_parallel_group, get_tensor_model_parallel_group
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.tensor_parallel.random import model_parallel_cuda_manual_seed
 from megatron.core.transformer.attention import SelfAttention, SelfAttentionSubmodules
 from megatron.core.transformer.dot_product_attention import DotProductAttention
@@ -66,7 +66,7 @@ class TestSpecCustomization:
         )
 
         # Create model process groups for test.
-        self.pg_collection = ProcessGroupCollection(
+        self.model_comm_pgs = ModelCommProcessGroups(
             tp=get_tensor_model_parallel_group(), cp=get_context_parallel_group()
         )
 
@@ -150,8 +150,22 @@ class TestSpecCustomization:
             hidden_size=12,
             num_attention_heads=4,
             use_cpu_initialization=True,
-            window_size=(10, 0),
+            window_size=[10, 0],
         )
+        # Make sure DotProductAttention throws (swa unsupported).
+        threw = False
+        try:
+            attn = DotProductAttention(
+                config,
+                layer_number=1,
+                attn_mask_type=AttnMaskType.causal,
+                attention_type='self',
+                model_comm_pgs=self.model_comm_pgs,
+            )
+        except:
+            threw = True
+        finally:
+            assert threw, 'Expected DotProductAttention to throw exception for SWA'
 
         # Test TEDotProductAttention
         attn = TEDotProductAttention(
@@ -159,7 +173,7 @@ class TestSpecCustomization:
             layer_number=1,
             attn_mask_type=AttnMaskType.causal,
             attention_type='self',
-            pg_collection=self.pg_collection,
+            model_comm_pgs=self.model_comm_pgs,
         )
         # Make sure window-size is what we expect.
         assert attn.window_size == config.window_size
@@ -173,7 +187,7 @@ class TestSpecCustomization:
                 layer_number=1,
                 attn_mask_type=AttnMaskType.causal,
                 attention_type='self',
-                pg_collection=self.pg_collection,
+                model_comm_pgs=self.model_comm_pgs,
             )
         except:
             threw = True
@@ -187,7 +201,7 @@ class TestSpecCustomization:
             layer_number=1,
             attn_mask_type=AttnMaskType.causal,
             attention_type='self',
-            pg_collection=self.pg_collection,
+            model_comm_pgs=self.model_comm_pgs,
         )
         # Make sure it's causal.
         assert attn.window_size == (-1, 0)

