diff --git a/megatron/core/models/T5/t5_model.py b/megatron/core/models/T5/t5_model.py
index 6d134067c..d8aa7aed3 100644
--- a/megatron/core/models/T5/t5_model.py
+++ b/megatron/core/models/T5/t5_model.py
@@ -5,7 +5,7 @@ from typing import List, Literal, Optional, Tuple
 import torch
 from torch import Tensor
 
-from megatron.core import tensor_parallel
+from megatron.core import parallel_state, tensor_parallel
 from megatron.core.config_logger import has_config_logger_enabled, log_config_to_disk
 from megatron.core.dist_checkpointing.mapping import ShardedStateDict
 from megatron.core.enums import ModelType
@@ -15,7 +15,7 @@ from megatron.core.models.common.embeddings.relative_pos_embedding import Relati
 from megatron.core.models.common.embeddings.rotary_pos_embedding import RotaryEmbedding
 from megatron.core.models.common.language_module.language_module import LanguageModule
 from megatron.core.packed_seq_params import PackedSeqParams
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.tensor_parallel.mappings import scatter_to_tensor_model_parallel_region
 from megatron.core.transformer.module import MegatronModule
 from megatron.core.transformer.spec_utils import ModuleSpec
@@ -151,7 +151,7 @@ class T5Model(LanguageModule):
         relative_attention_max_distance: int = 128,
         add_encoder: bool = True,
         add_decoder: bool = True,
-        pg_collection: ProcessGroupCollection = None,
+        model_comm_pgs: ModelCommProcessGroups = None,
     ):
 
         super(T5Model, self).__init__(config=config)
@@ -171,13 +171,13 @@ class T5Model(LanguageModule):
         self.share_embeddings_and_output_weights = share_embeddings_and_output_weights
         self.position_embedding_type = position_embedding_type
         self.encoder_hidden_state = None
-        if pg_collection is None:
-            pg_collection = ProcessGroupCollection.use_mpu_process_groups(
+        if model_comm_pgs is None:
+            model_comm_pgs = ModelCommProcessGroups.use_mpu_process_groups(
                 required_pgs=['tp', 'cp', 'pp']
             )
-        self.tp_group = get_tensor_model_parallel_group_if_none(pg_collection.tp)
+        self.tp_group = get_tensor_model_parallel_group_if_none(model_comm_pgs.tp)
 
-        self.model_type = ModelType.encoder_or_decoder
+        self.model_type = ModelType.encoder_and_decoder
 
         # Tells schedules.py that this model has a skip connection
         # between the encoder's output and the decoder
@@ -209,7 +209,7 @@ class T5Model(LanguageModule):
                 rotary_interleaved=self.config.rotary_interleaved,
                 seq_len_interpolation_factor=seq_len_interpolation_factor,
                 use_cpu_initialization=self.config.use_cpu_initialization,
-                cp_group=pg_collection.cp,
+                cp_group=model_comm_pgs.cp,
             )
 
         # Relative Position Embeddings
@@ -240,7 +240,7 @@ class T5Model(LanguageModule):
                 spec=encoder_spec,
                 pre_process=self.pre_process,
                 post_process=self.post_process,
-                pg_collection=pg_collection,
+                model_comm_pgs=model_comm_pgs,
             )
         else:
             self.encoder = None
@@ -252,7 +252,7 @@ class T5Model(LanguageModule):
                 spec=decoder_spec,
                 pre_process=self.pre_process,
                 post_process=self.post_process,
-                pg_collection=pg_collection,
+                model_comm_pgs=model_comm_pgs,
             )
         else:
             self.decoder = None
@@ -494,6 +494,11 @@ class T5Model(LanguageModule):
             ShardedStateDict: sharded state dict for the T5Model
         """
         sharded_sd = super().sharded_state_dict(prefix, sharded_offsets, metadata)
+        if not parallel_state.is_inside_encoder():
+            for k, sh_ten in sharded_sd.items():
+                if not k.startswith(f'{prefix}decoder'):
+                    # Bump replica_id of all the layers shared with the encoder (output, embedding)
+                    sh_ten.replica_id = (sh_ten.replica_id[0] + 1, *sh_ten.replica_id[1:])
         return sharded_sd
 
 

