diff --git a/tests/unit_tests/ssm/test_mamba_layer.py b/tests/unit_tests/ssm/test_mamba_layer.py
index 25d24443d..59f9f832f 100644
--- a/tests/unit_tests/ssm/test_mamba_layer.py
+++ b/tests/unit_tests/ssm/test_mamba_layer.py
@@ -4,7 +4,7 @@ import pytest
 import torch
 
 from megatron.core.models.mamba.mamba_layer_specs import mamba_stack_spec
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.ssm.mamba_layer import MambaLayer
 from megatron.core.tensor_parallel.random import model_parallel_cuda_manual_seed
 from megatron.core.transformer import TransformerConfig
@@ -26,8 +26,8 @@ class TestMambaLayer:
             use_cpu_initialization=True,
         )
         modules = mamba_stack_spec.submodules.mamba_layer.submodules
-        pg_collection = ProcessGroupCollection.use_mpu_process_groups(required_pgs=['tp', 'cp'])
-        self.layer = MambaLayer(transformer_config, modules, pg_collection=pg_collection)
+        model_comm_pgs = ModelCommProcessGroups.use_mpu_process_groups(required_pgs=['tp', 'cp'])
+        self.layer = MambaLayer(transformer_config, modules, model_comm_pgs=model_comm_pgs)
 
     def teardown_method(self, method):
         Utils.destroy_model_parallel()

