diff --git a/megatron/core/tokenizers/text/models/__init__.py b/megatron/core/tokenizers/text/models/__init__.py
deleted file mode 100644
index 3610a8b98..000000000
--- a/megatron/core/tokenizers/text/models/__init__.py
+++ /dev/null
@@ -1,8 +0,0 @@
-# Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
-
-from megatron.core.tokenizers.text.models.bert_tokenizer import BertTokenizer
-from megatron.core.tokenizers.text.models.default_tokenizer import DefaultTokenizerText
-from megatron.core.tokenizers.text.models.gpt_tokenizer import GPTTokenizer
-from megatron.core.tokenizers.text.models.mamba_tokenizer import MambaTokenizer
-from megatron.core.tokenizers.text.models.retro_tokenizer import RetroTokenizer
-from megatron.core.tokenizers.text.models.t5_tokenizer import T5Tokenizer

