diff --git a/tests/unit_tests/distributed/test_grad_reduce_for_replicated_embedder.py b/tests/unit_tests/distributed/test_grad_reduce_for_replicated_embedder.py
index c5acf0b76..8028c041c 100644
--- a/tests/unit_tests/distributed/test_grad_reduce_for_replicated_embedder.py
+++ b/tests/unit_tests/distributed/test_grad_reduce_for_replicated_embedder.py
@@ -31,9 +31,7 @@ def test_allreduce_conditional_embedding_grads():
         for param in chunk.parameters():
             param.main_grad = torch.ones_like(param) * (pp_rank * 10.0 + i)
 
-    _allreduce_conditional_embedding_grads(
-        model, config, parallel_state.get_pipeline_model_parallel_group()
-    )
+    _allreduce_conditional_embedding_grads(model, config)
 
     expect_value = 0
     for i in range(len(model)):

