diff --git a/LICENSE b/LICENSE
index 621920c37..aa248579f 100644
--- a/LICENSE
+++ b/LICENSE
@@ -37,7 +37,7 @@ Below are licenses used in those files, as indicated.
 
 
 --------------------------------------------------------------------------------------
--- LICENSE FOR Facebook, huggingface, Google Research, LLaVA, Mamba, TinyZero and vLLM code  --
+-- LICENSE FOR Facebook, huggingface, Google Research, LLaVA, Mamba, and vLLM code  --
 
 
                                  Apache License

