diff --git a/pyproject.toml b/pyproject.toml
index 6a1d5d26c..3331114d7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,7 +2,6 @@
 
 [build-system]
 requires = ["setuptools<80.0.0", "pybind11"]
-build-backend = "setuptools.build_meta"
 
 [tool.setuptools]
 include-package-data = true
@@ -20,7 +19,7 @@ dynamic = ["version", "readme"]
 description = "Megatron Core - a library for efficient and scalable training of transformer based models"
 requires-python = ">=3.10"
 license = { text = "Apache 2.0" }
-dependencies = ["torch", "numpy<2.0.0", "packaging>=24.2"]
+dependencies = ["torch", "numpy<2.0.0"]
 authors = [{ name = "NVIDIA", email = "nemo-toolkit@nvidia.com" }]
 maintainers = [{ name = "NVIDIA", email = "nemo-toolkit@nvidia.com" }]
 keywords = [
@@ -63,26 +62,18 @@ Download = "https://github.com/NVIDIA/Megatron-LM/releases"
 Homepage = "https://github.com/NVIDIA/Megatron-LM/megatron/core"
 
 [project.optional-dependencies]
-mlm = ["flask-restful", "sentencepiece", "tiktoken", "wandb", "transformers"]
+mlm = ["flask-restful", "sentencepiece", "tiktoken", "wandb"]
 
 dev = [
     "tqdm",
-    "einops~=0.8",
-    "tensorstore~=0.1,!=0.1.46,!=0.1.72",
-    "nvtx~=0.2",
-    "multi-storage-client~=0.27",
-    "opentelemetry-api~=1.33.1",
+    "einops",
+    "tensorstore!=0.1.46,!=0.1.72",
+    "nvtx",
+    "transformers",
+    "multi-storage-client",
     "setuptools<80.0.0",
-    "mamba-ssm~=2.2",
-    "causal-conv1d~=1.5",
-    "nv-grouped-gemm~=1.1",
-    "transformer-engine[pytorch]>=2.6.0a0,<2.8.0",
-    "nvidia-resiliency-ext>=0.4.0a0,<0.5.0",
-    "nvidia-modelopt[torch]>=0.33.0a0,<0.34.0; sys_platform != 'darwin'",
-    "megatron-energon[av_decode]~=6.0",
-    "flashinfer-python",
-    "wget",
-    "onnxscript",
+    "nvidia-modelopt[torch]; sys_platform != 'darwin'",
+    "megatron-energon[av_decode]<7",
 ]
 
 lts = [
@@ -93,56 +84,34 @@ lts = [
     "transformers",
     "zarr",
     "setuptools<80.0.0",
-    "wget",
 ]
 
 [dependency-groups]
 test = [
-    "coverage",
     "nltk",
     "wrapt",
-    "pytest==8.3.5",
+    "pytest",
     "pytest-mock",
     "pytest-cov",
     "pytest-random-order",
     "pytest-asyncio",
-    "pygithub",
-    "pydantic",
-    "tensorboard",
-    "pyyaml",
 ]
 
 build = [
     "setuptools<80.0.0",
-    "packaging>=24.2",
+    "packaging",
     "hatchling",
     "pybind11",
     "Cython>=3.0.0",
-    "torch",
 ]
-linting = [
-    "ruff~=0.9.0",
-    "black==24.4.2",
-    "isort==5.13.2",
-    "flake8==7.1.0",
-    "pylint==3.2.6",
-]
-ci = ["python-gitlab", "slack-sdk", "pandas"]
-flash_mla = ["flash_mla"]
+linting = ["ruff~=0.9.0"]
 
 [tool.uv]
 default-groups = ["linting", "build", "test"]
-no-build-isolation-package = [
-    "transformer-engine",
-    "transformer-engine-torch",
-    "mamba-ssm",
-    "causal-conv1d",
-    "nv-grouped-gemm",
-    "flash_mla",
-]
+
 link-mode = "copy"
 conflicts = [[{ extra = "lts" }, { extra = "dev" }]]
-# We don't want to install torch, torchvision, and triton
+# We don't want to install torch, torchvision, and triton 
 # because they are already installed in the base image.
 override-dependencies = [
     "torch; sys_platform == 'never'",
@@ -150,12 +119,6 @@ override-dependencies = [
     "triton; sys_platform == 'never'",
 ]
 
-[tool.uv.sources]
-flash_mla = [
-    { git = "https://github.com/deepseek-ai/FlashMLA", rev = "9edee0c022cd0938148a18e334203b0aab43aa19" },
-]
-transformer-engine = { git = "https://github.com/NVIDIA/TransformerEngine.git", rev = "0289e76380088358a584d809faf69effab1a7cda" } # on `release_v2.7
-
 [tool.isort]
 profile = "black"                                                          # black-compatible
 line_length = 100                                                          # should match black parameters
@@ -208,8 +171,7 @@ source = ["/opt/megatron-lm/"]
 [tool.ruff.lint]
 # Enable all `pydocstyle` rules, limiting to those that adhere to the
 # Google convention via `convention = "google"`, below.
-# select = ["D", "F"]
-select = ["S506"]
+select = ["D", "F"]
 
 # - On top of the Google convention, disable `D417`, which requires
 #   documentation for every function parameter.

