diff --git a/tools/autoformat.sh b/tools/autoformat.sh
index 6c3e76b3e..ed4469b6a 100755
--- a/tools/autoformat.sh
+++ b/tools/autoformat.sh
@@ -19,14 +19,10 @@ CHANGED_FILES=$(git diff --name-only --diff-filter=d --merge-base origin/${BASE_
 ADDITIONAL_ARGS=""
 ADDITIONAL_BLACK_ARGS=""
 ADDITIONAL_PYLINT_ARGS=""
-ADDITIONAL_RUFF_ARGS=""
 
 if [[ $CHECK_ONLY == true ]]; then
     ADDITIONAL_ARGS="--check"
     ADDITIONAL_BLACK_ARGS="--diff"
-    ADDITIONAL_RUFF_ARGS="--no-fix"
-else
-    ADDITIONAL_RUFF_ARGS="--fix"
 fi
 
 if [[ $SKIP_DOCS == true ]]; then
@@ -37,7 +33,6 @@ if [[ -n "$CHANGED_FILES" ]]; then
     black --skip-magic-trailing-comma --skip-string-normalization $ADDITIONAL_ARGS $ADDITIONAL_BLACK_ARGS --verbose $CHANGED_FILES
     isort $ADDITIONAL_ARGS $CHANGED_FILES
     pylint $ADDITIONAL_PYLINT_ARGS $CHANGED_FILES
-    ruff check $ADDITIONAL_RUFF_ARGS $CHANGED_FILES
     mypy --explicit-package-bases --follow-imports=skip $CHANGED_FILES || true
 else
     echo Changeset is empty, all good.

