diff --git a/tools/preprocess_data.py b/tools/preprocess_data.py
index 7d382a0d1..c90feb5c1 100644
--- a/tools/preprocess_data.py
+++ b/tools/preprocess_data.py
@@ -12,6 +12,7 @@ import time
 import gzip
 import glob
 import torch
+import shutil
 import numpy as np
 import multiprocessing
 try:
@@ -22,8 +23,6 @@ except ImportError:
     PunktLanguageVars = object  # Fallback to the built-in object class
     nltk_available = False
 
-from megatron.core.tokenizers.text.utils.build_tokenizer import build_tokenizer as build_new_tokenizer
-from megatron.core.tokenizers import MegatronTokenizer
 from megatron.training.tokenizer import build_tokenizer
 from megatron.training.arguments import _add_tokenizer_args
 from megatron.core.datasets import indexed_dataset
@@ -53,10 +52,7 @@ class Encoder(object):
 
     def initializer(self):
         # Use Encoder class as a container for global data
-        if self.args.legacy_tokenizer:
-            tokenizer = build_tokenizer(self.args)
-        else:
-            Encoder.tokenizer = build_new_tokenizer(self.args)
+        Encoder.tokenizer = build_tokenizer(self.args)
         if self.args.split_sentences:
             if not nltk_available:
                 print("NLTK is not available to split sentences.")
@@ -156,10 +152,7 @@ class Partition(object):
 
         startup_start = time.time()
         encoder = Encoder(self.args)
-        if self.args.legacy_tokenizer:
-            tokenizer = build_tokenizer(self.args)
-        else:
-            tokenizer = build_new_tokenizer(self.args)
+        tokenizer = build_tokenizer(self.args)
         pool = multiprocessing.Pool(self.workers, initializer=encoder.initializer)
         encoded_docs = pool.imap(encoder.encode, fin, 32)
 
@@ -227,8 +220,6 @@ def get_args():
     group.add_argument('--keep-sequential-samples', action='store_true',
                        help='Ensure ordering of samples in .jsonl files is '
                             'preserved when using partitions>1.')
-    # group.add_argument('--legacy-tokenizer', action='store_true',
-    #                    help='Use legacy tokenizer system.')
     args = parser.parse_args()
     args.keep_empty = False
 
@@ -268,7 +259,12 @@ def main():
 
     if args.split_sentences:
         if nltk_available:
-            nltk.download("punkt", quiet=True, download_dir=os.environ.get("NLTK_DATA"))
+            try:
+                punkt_path = os.environ.get("NLTK_DATA") + "/tokenizers/punkt"
+                if not os.path.exists(punkt_path):
+                    shutil.copytree('/root/nltk_data/tokenizers/punkt', punkt_path)
+            except:
+                nltk.download("punkt", quiet=True, download_dir=os.environ.get("NLTK_DATA"))
         else:
             raise Exception(
                 "nltk library required for sentence splitting is not available.")
@@ -381,10 +377,7 @@ def main():
     output_bin_files = {}
     output_idx_files = {}
     builders = {}
-    if args.legacy_tokenizer:
-        tokenizer = build_tokenizer(args)
-    else:
-        tokenizer = build_new_tokenizer(args)
+    tokenizer = build_tokenizer(args)
 
     for key in args.json_keys:
         output_bin_files[key] = "{}_{}_{}.bin".format(args.output_prefix,

