diff --git a/tools/run_inference_performance_test.py b/tools/run_inference_performance_test.py
index 17db26a5d..554c63999 100644
--- a/tools/run_inference_performance_test.py
+++ b/tools/run_inference_performance_test.py
@@ -3,15 +3,14 @@ from megatron.core.inference.model_inference_wrappers.inference_wrapper_config i
     InferenceWrapperConfig,
 )
 import argparse
+from pretrain_gpt import model_provider as gpt_model_provider
+from pretrain_mamba import model_provider as mamba_model_provider
 import random
 import torch
 import sys
 import time
 import tqdm
 import warnings
-from model_provider import model_provider
-from gpt_builders import gpt_builder
-from mamba_builders import mamba_builder
 from megatron.core.inference.engines.abstract_engine import AbstractEngine
 from megatron.core.inference.engines import DynamicInferenceEngine, StaticInferenceEngine
 from megatron.core.inference.inference_request import InferenceRequest
@@ -245,8 +244,9 @@ def generate_dynamic(
     start_time = time.perf_counter()
     all_finished_requests = []
     while inference_engine.has_unfinished_requests():
-        result = inference_engine.step(sampling_params, verbose=False)
-        finished_requests = result["finished_requests"]
+        active_requests, finished_requests, step_time = inference_engine.step(
+            sampling_params, verbose=False
+        )
         for request in finished_requests:
             req_id = request.request_id
             latency = time.perf_counter() - start_time
@@ -279,11 +279,11 @@ def main():
 
     # Set up model and load checkpoint
     if args.model_provider == "gpt":
-        model_builder = gpt_builder
+        model_provider = gpt_model_provider
     elif args.model_provider == "mamba":
-        model_builder = mamba_builder
+        model_provider = mamba_model_provider
 
-    model = get_model(model_provider(model_builder), wrap_with_ddp=False)
+    model = get_model(model_provider, wrap_with_ddp=False)
     tokenizer = get_tokenizer()
     load_checkpoint(model, None, None)
     model = model[0]

