diff --git a/tests/unit_tests/tensor_parallel/test_random.py b/tests/unit_tests/tensor_parallel/test_random.py
index 47b607b87..48273003c 100644
--- a/tests/unit_tests/tensor_parallel/test_random.py
+++ b/tests/unit_tests/tensor_parallel/test_random.py
@@ -35,7 +35,7 @@ def test_cuda_rng_states_tracker():
 
 def test_model_parallel_cuda_manual_seed():
     Utils.initialize_model_parallel(4, 2)
-    model_parallel_cuda_manual_seed(0, force_reset_rng=True)
+    model_parallel_cuda_manual_seed(0)
     rng_tracker = get_cuda_rng_tracker()
     assert rng_tracker.get_states()['model-parallel-rng'] is not None
     Utils.destroy_model_parallel()

