diff --git a/megatron/training/tokenizer/sft_tokenizer.py b/megatron/training/tokenizer/sft_tokenizer.py
index 4a941fc18..0efc39083 100644
--- a/megatron/training/tokenizer/sft_tokenizer.py
+++ b/megatron/training/tokenizer/sft_tokenizer.py
@@ -5,13 +5,12 @@ from typing import Dict, List, Union
 import numpy as np
 
 nemotron_h_aligned_custom_template = """{% for message in messages %}{% if message['role'] == 'system' %}{{ '<SPECIAL_10>System\n' + message['content'].strip() + '\n' }}{% elif message['role'] == 'user' %}{{ '<SPECIAL_11>User\n' + message['content'].strip() + '\n' + '<SPECIAL_11>Assistant\n' }}{% elif message['role'] == 'assistant' %}{{ message['content'].strip() + '\n' }}{% endif %}{% endfor %}"""
-nemotron_nano_v2_custom_template = """{% for message in messages %}{% set content = message['content'] %}{% if message['role'] == 'system' %}{{ '<SPECIAL_10>System\n' + content.replace('/think', '').replace('/no_think', '').strip() + '\n' }}{% elif message['role'] == 'user' %}{{ '<SPECIAL_11>User\n' + content.replace('/think', '').replace('/no_think', '').strip() + '\n' }}{% elif message['role'] == 'assistant' %}{{ '<SPECIAL_11>Assistant\n' + content.strip() + '\n<SPECIAL_12>\n' }}{% endif %}{% endfor %}"""
 
-from megatron.core.datasets.megatron_tokenizer import MegatronLegacyTokenizer
+from megatron.core.datasets.megatron_tokenizer import MegatronTokenizer
 from megatron.training.datasets.sft_dataset import IGNORE_INDEX
 from megatron.training.tokenizer.multimodal_tokenizer import PromptConfig
 
-class SFTTokenizer(MegatronLegacyTokenizer):  
+class SFTTokenizer(MegatronTokenizer):  
     """SFT Tokenizer."""
 
     def __init__(
@@ -42,15 +41,7 @@ class SFTTokenizer(MegatronLegacyTokenizer):
         self._vocab_size = len(tokenizer)
         self._tokenizer = tokenizer
 
-        if prompt_format == "nemotron-nano-v2":
-            self._prompt_config = PromptConfig(
-                assistant_prefix_len=3,
-                pad_token_id=tokenizer.convert_tokens_to_ids("<unk>"),
-                custom_chat_template=nemotron_nano_v2_custom_template,
-                has_bos=False,
-                has_system_role=True,
-            )
-        elif prompt_format == "nemotron-h-aligned":
+        if prompt_format == "nemotron-h-aligned":
             self._prompt_config = PromptConfig(
                 assistant_prefix_len=0,
                 pad_token_id=tokenizer.convert_tokens_to_ids("<SPECIAL_233>"),
@@ -160,11 +151,6 @@ class SFTTokenizer(MegatronLegacyTokenizer):
         """Get special tokens."""
         return self._tokenizer.get_added_vocab()
 
-    @property
-    def force_eod(self):
-        """To force an EOD at the end of every data sample in SFT."""
-        return self._prompt_format == "nemotron-h-aligned"
-
     @property
     def pad(self):
         """Pad token ID."""

