diff --git a/tests/functional_tests/python_test_utils/conftest.py b/tests/functional_tests/python_test_utils/conftest.py
index 7cf6cd4e5..b97e1961c 100644
--- a/tests/functional_tests/python_test_utils/conftest.py
+++ b/tests/functional_tests/python_test_utils/conftest.py
@@ -14,12 +14,6 @@ def pytest_addoption(parser):
         help="If set, test system checks for approximate results.",
     )
     parser.addoption("--golden-values-path", action="store", help="Path to golden values")
-    parser.addoption("--actual-values-path", action="store", help="Path to actual values")
-    parser.addoption("--actual-values-first-run-path", action="store", help="Path to actual values")
-    parser.addoption(
-        "--actual-values-second-run-path", action="store", help="Path to actual values"
-    )
-    parser.addoption("--scope", action="store", help="Test scope (MR, weekly, prerelease, release)")
     parser.addoption(
         "--train-iters", action="store", default=100, help="Number of train iters", type=int
     )
@@ -46,34 +40,6 @@ def golden_values(request):
     return common.read_golden_values_from_json(request.config.getoption("--golden-values-path"))
 
 
-@pytest.fixture
-def actual_values(request):
-    """Simple fixture returning golden values."""
-    return common.read_golden_values_from_json(request.config.getoption("--actual-values-path"))
-
-
-@pytest.fixture
-def actual_values_first_run(request):
-    """Simple fixture returning actual values."""
-    return common.read_golden_values_from_json(
-        request.config.getoption("--actual-values-first-run-path")
-    )
-
-
-@pytest.fixture
-def actual_values_second_run(request):
-    """Simple fixture returning actual values."""
-    return common.read_golden_values_from_json(
-        request.config.getoption("--actual-values-second-run-path")
-    )
-
-
-@pytest.fixture
-def scope(request):
-    """Simple fixture returning golden values."""
-    return request.config.getoption("--scope")
-
-
 @pytest.fixture
 def train_iters(request):
     """Simple fixture returning number of train iters."""

