diff --git a/tests/functional_tests/python_test_utils/test_pretraining_regular_pipeline.py b/tests/functional_tests/python_test_utils/test_pretraining_regular_pipeline.py
index 2887a65df..761b84d01 100644
--- a/tests/functional_tests/python_test_utils/test_pretraining_regular_pipeline.py
+++ b/tests/functional_tests/python_test_utils/test_pretraining_regular_pipeline.py
@@ -20,10 +20,6 @@ CHECK_THRESHOLDS = {
         common.DeterministicTest(),
         common.ApproximateTest(atol_func=common.approximate_threshold(rtol=0.05), rtol=0),
     ],
-    "mtp_1 loss": [
-        common.DeterministicTest(),
-        common.ApproximateTest(atol_func=common.approximate_threshold(rtol=0.05), rtol=0),
-    ],
     "num-zeros": [
         common.DeterministicTest(),
         common.ApproximateTest(atol_func=common.approximate_threshold(rtol=0.20), rtol=0),
@@ -42,7 +38,7 @@ CHECK_THRESHOLDS = {
 def test_regular_pipeline(
     compare_approximate_results: bool,
     golden_values: Dict[str, common.GoldenValueMetric],
-    actual_values: Dict[str, common.GoldenValueMetric],
+    tensorboard_logs: Dict[str, common.GoldenValueMetric],
     model_config_path: str,
     checks: Optional[Dict[str, List[common.Test]]] = None,
 ):
@@ -75,6 +71,6 @@ def test_regular_pipeline(
     common.pipeline(
         compare_approximate_results=compare_approximate_results,
         golden_values=golden_values,
-        actual_values=actual_values,
+        tensorboard_logs=tensorboard_logs,
         checks=checks,
     )

