diff --git a/megatron/core/post_training/modelopt/gpt/state_dict_hooks.py b/megatron/core/post_training/modelopt/gpt/state_dict_hooks.py
index 596d210f5..bf1d86af1 100644
--- a/megatron/core/post_training/modelopt/gpt/state_dict_hooks.py
+++ b/megatron/core/post_training/modelopt/gpt/state_dict_hooks.py
@@ -47,7 +47,6 @@ def mcore_gpt_load_te_state_dict_pre_hook(
         ("self_attention.linear_kv_up_proj.layer_norm_bias", "self_attention.kv_layernorm.bias"),
         ("mlp.linear_fc1.layer_norm_weight", "pre_mlp_layernorm.weight"),
         ("mlp.linear_fc1.layer_norm_bias", "pre_mlp_layernorm.bias"),
-        ("mixer.in_proj.layer_norm_weight", "norm.weight"),
     ]
 
     key_rewrite_list = []

