diff --git a/tests/test_utils/python_scripts/check_status_of_main.py b/tests/test_utils/python_scripts/check_status_of_main.py
index 5e7ef5171..db6d6438d 100644
--- a/tests/test_utils/python_scripts/check_status_of_main.py
+++ b/tests/test_utils/python_scripts/check_status_of_main.py
@@ -46,14 +46,13 @@ def is_pending(target_branch: str):
     is_pending = (
         pipeline.attributes['status'] == 'pending' or pipeline.attributes['status'] == 'running'
     )
-    is_canceled = pipeline.attributes['status'] == 'canceled'
 
     if not is_pending:
         logger.info(
             f"Main pipeline {pipeline.id} finished with status {pipeline.attributes['status']}"
         )
 
-    return is_pending or is_canceled
+    return is_pending
 
 
 def is_sucess(target_branch: str):

