diff --git a/tests/test_utils/python_scripts/generate_jet_trigger_job.py b/tests/test_utils/python_scripts/generate_jet_trigger_job.py
index 6c5ff73c3..3fe0ef713 100644
--- a/tests/test_utils/python_scripts/generate_jet_trigger_job.py
+++ b/tests/test_utils/python_scripts/generate_jet_trigger_job.py
@@ -131,7 +131,7 @@ def main(
                 "rules": [
                     {
                         "if": '($CI_PIPELINE_SOURCE == "parent_pipeline" || $CI_MERGE_REQUEST_ID) && $CI_COMMIT_BRANCH == "main"',
-                        "auto_cancel": {"on_new_commit": "interruptible"},
+                        "auto_cancel": {"on_new_commit": "none"},
                     },
                     {"if": '$CI_PIPELINE_SOURCE == "parent_pipeline" || $CI_MERGE_REQUEST_ID'},
                     {"when": "never"},

