diff --git a/tests/test_utils/python_scripts/launch_jet_workload.py b/tests/test_utils/python_scripts/launch_jet_workload.py
index d8aa0d80c..5a47ffe2d 100644
--- a/tests/test_utils/python_scripts/launch_jet_workload.py
+++ b/tests/test_utils/python_scripts/launch_jet_workload.py
@@ -116,10 +116,7 @@ def launch_and_wait_for_completion(
                 wait_for_validation=True,
                 max_wait_time=(60 * 60),
             )
-        except (
-            jetclient.clients.gitlab.GitlabAPIError,
-            jetclient.facades.objects.util.WaitTimeExceeded,
-        ) as e:
+        except jetclient.clients.gitlab.GitlabAPIError as e:
             logger.error(f"Faced {str(e)}. Waiting and retrying...")
             n_submission_attempts += 1
             time.sleep(2**n_submission_attempts * 5)
@@ -270,14 +267,11 @@ def is_flaky_failure(concat_allranks_logs: str) -> bool:
         or "illegal instruction" in concat_allranks_logs
         or "torch.distributed.DistNetworkError" in concat_allranks_logs
         or "Segmentation fault" in concat_allranks_logs
-        or "found NaN in" in concat_allranks_logs
+        or "found NaN in local forward loss calculation" in concat_allranks_logs
         or "For debugging consider passing CUDA_LAUNCH_BLOCKING=1" in concat_allranks_logs
         or "double free or corruption" in concat_allranks_logs
         or "Call to CUDA function failed." in concat_allranks_logs
         or "Connection reset by peer" in concat_allranks_logs
-        or "invalid pointer" in concat_allranks_logs
-        or "malloc(): unaligned tcache chunk detected" in concat_allranks_logs
-        or "zmq.error.ZMQError: Address already in use" in concat_allranks_logs
     )
 
 
@@ -375,7 +369,7 @@ def main(
     n_attempts = 0
     n_nondeterminism_attemps = 0
     n_iteration = 0
-    while True and n_attempts < 9 and n_nondeterminism_attemps < 3:
+    while True and n_attempts < 3 and n_nondeterminism_attemps < 2:
         pipeline = launch_and_wait_for_completion(
             test_case=test_case,
             environment=environment,
@@ -483,9 +477,7 @@ def main(
                 n_attempts += 1
                 continue
 
-            if (
-                "FAILED tests/functional_tests/python_test_utils" in concat_mainrank_log
-            ) and re.compile(r"\bEXIT_CODE=0\b").search(concat_mainrank_log) is not None:
+            if "FAILED tests/functional_tests/python_test_utils" in concat_mainrank_log:
                 logger.error("Non-determinism, let's try another node.")
                 n_nondeterminism_attemps += 1
                 continue
@@ -502,10 +494,9 @@ def main(
             if (
                 "StopIteration" in concat_allranks_logs
                 or "after training is done" in concat_allranks_logs
-                or "exiting program at iteration" in concat_allranks_logs
             ):
                 logger.info("Release training finished")
-                sys.exit(int(not success))  # invert for exit 0
+                sys.exit(0)
 
             if parse_failed_job(logs=mainrank_log):
                 n_attempts += 1

