diff --git a/megatron/core/ssm/mamba_layer.py b/megatron/core/ssm/mamba_layer.py
index 13f4aca51..d1946de8e 100644
--- a/megatron/core/ssm/mamba_layer.py
+++ b/megatron/core/ssm/mamba_layer.py
@@ -14,7 +14,7 @@ from torch import Tensor
 from megatron.core.dist_checkpointing.mapping import ShardedStateDict
 from megatron.core.dist_checkpointing.utils import apply_prefix_mapping
 from megatron.core.inference.contexts import BaseInferenceContext
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.transformer.cuda_graphs import CudaGraphManager
 from megatron.core.transformer.identity_op import IdentityOp
 from megatron.core.transformer.module import MegatronModule
@@ -61,13 +61,13 @@ class MambaLayer(MegatronModule):
         submodules: MambaLayerSubmodules,
         layer_number: int = 1,
         residual_in_fp32=False,
-        pg_collection: ProcessGroupCollection = None,
+        model_comm_pgs: ModelCommProcessGroups = None,
     ):
         """Initialize Mamba Layer."""
         super().__init__(config)
-        assert pg_collection is not None, "pg_collection must be provided for MambaLayer"
+        assert model_comm_pgs is not None, "model_comm_pgs must be provided for MambaLayer"
 
-        if config.enable_cuda_graph and config.cuda_graph_scope != "full_iteration":
+        if config.enable_cuda_graph:
             self.cudagraph_manager = CudaGraphManager(config)
 
         self.config = config
@@ -80,7 +80,7 @@ class MambaLayer(MegatronModule):
             self.config,
             d_model=self.config.hidden_size,
             layer_number=layer_number,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )
         self.norm = build_module(submodules.norm, self.config, self.config.hidden_size)
         self.mamba_bda = build_module(submodules.mamba_bda)

