diff --git a/megatron/core/ssm/mlp_layer.py b/megatron/core/ssm/mlp_layer.py
index 19aec5878..7ab1f76e2 100644
--- a/megatron/core/ssm/mlp_layer.py
+++ b/megatron/core/ssm/mlp_layer.py
@@ -2,7 +2,7 @@
 
 from typing import Optional
 
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.transformer import (
     TransformerConfig,
     TransformerLayer,
@@ -19,12 +19,12 @@ class MLPLayer(TransformerLayer):
         submodules: TransformerLayerSubmodules,
         layer_number: int = 1,
         hidden_dropout: float = None,
-        pg_collection: Optional[ProcessGroupCollection] = None,
+        model_comm_pgs: Optional[ModelCommProcessGroups] = None,
     ):
         super().__init__(
             config=config,
             submodules=submodules,
             layer_number=layer_number,
             hidden_dropout=hidden_dropout,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )

