diff --git a/megatron/core/transformer/attention.py b/megatron/core/transformer/attention.py
index 6b2e898be..6ebaacedd 100644
--- a/megatron/core/transformer/attention.py
+++ b/megatron/core/transformer/attention.py
@@ -22,15 +22,13 @@ from megatron.core.parallel_state import (
     get_tensor_model_parallel_rank,
     get_tensor_model_parallel_world_size,
 )
-from megatron.core.process_groups_config import ProcessGroupCollection
+from megatron.core.process_groups_config import ModelCommProcessGroups
 from megatron.core.transformer.module import MegatronModule
 from megatron.core.transformer.spec_utils import ModuleSpec, build_module
 from megatron.core.utils import (
     deprecate_inference_params,
     divide,
-    get_pg_size,
     is_fa_min_version,
-    is_te_min_version,
     nvtx_range_pop,
     nvtx_range_push,
 )
@@ -53,17 +51,6 @@ try:
 except:
     HAVE_FA3 = False
 
-try:
-    from flash_mla import flash_mla_with_kvcache, get_mla_metadata
-
-    HAVE_FMLA = True
-except ImportError:
-    flash_mla_with_kvcache = None
-    get_mla_metadata = None
-    HAVE_FMLA = False
-
-from megatron.core.transformer.transformer_config import MLATransformerConfig
-
 try:
     from flash_attn import flash_attn_varlen_func, flash_attn_with_kvcache
 except:
@@ -74,14 +61,10 @@ try:
     import transformer_engine  # pylint: disable=unused-import
 
     HAVE_TE = True
-    from megatron.core.extensions.transformer_engine import (
-        SplitAlongDim,
-        TELinear,
-        set_save_original_input,
-    )
+    from megatron.core.extensions.transformer_engine import SplitAlongDim
 except ImportError:
     HAVE_TE = False
-    SplitAlongDim, TELinear, set_save_original_input = None, None, None
+    SplitAlongDim = None
 
 
 @dataclass
@@ -124,7 +107,7 @@ class Attention(MegatronModule, ABC):
         attn_mask_type: AttnMaskType,
         attention_type: str,
         cp_comm_type: str = None,
-        pg_collection: ProcessGroupCollection = None,
+        model_comm_pgs: ModelCommProcessGroups = None,
     ):
         super().__init__(config=config)
 
@@ -138,19 +121,21 @@ class Attention(MegatronModule, ABC):
         self.query_projection_size = self.config.kv_channels * self.config.num_attention_heads
         self.kv_projection_size = self.config.kv_channels * self.config.num_query_groups
 
-        if pg_collection is None:
-            pg_collection = ProcessGroupCollection.use_mpu_process_groups(required_pgs=['tp', 'cp'])
+        if model_comm_pgs is None:
+            model_comm_pgs = ModelCommProcessGroups.use_mpu_process_groups(
+                required_pgs=['tp', 'cp']
+            )
         else:
             assert hasattr(
-                pg_collection, 'tp'
-            ), "Attention pg_collection must have tp process group"
+                model_comm_pgs, 'tp'
+            ), "Attention model_comm_pgs must have tp process group"
             assert hasattr(
-                pg_collection, 'cp'
-            ), "Attention pg_collection must have cp process group"
-        self.pg_collection = pg_collection
+                model_comm_pgs, 'cp'
+            ), "Attention model_comm_pgs must have cp process group"
+        self.model_comm_pgs = model_comm_pgs
 
         # Per attention head and per partition values
-        world_size = get_pg_size(self.pg_collection.tp)
+        world_size = self.model_comm_pgs.tp.size()
         self.hidden_size_per_attention_head = divide(
             self.query_projection_size, self.config.num_attention_heads
         )
@@ -169,7 +154,7 @@ class Attention(MegatronModule, ABC):
             attention_type=self.attention_type,
             cp_comm_type=cp_comm_type,
             softmax_scale=self.config.softmax_scale,
-            pg_collection=self.pg_collection,
+            model_comm_pgs=self.model_comm_pgs,
         )
 
         self.checkpoint_core_attention = (
@@ -189,22 +174,9 @@ class Attention(MegatronModule, ABC):
             skip_bias_add=True,
             is_expert=False,
             tp_comm_buffer_name='proj',
-            tp_group=self.pg_collection.tp,
+            tp_group=self.model_comm_pgs.tp,
         )
 
-        if (
-            HAVE_TE
-            and self.config.fp8
-            and self.config.fp8_recipe != 'delayed'
-            and is_te_min_version("2.6.0dev0")
-            and isinstance(self.linear_proj, TELinear)
-        ):
-            # For fp8 training, the output of the fused core_attn is saved by itself, and
-            # linear_proj also saves the quantized tensor of this output. Here we set the
-            # linear_proj to save the original input tensors to avoid the extra memory usage of
-            # the quantized tensor.
-            set_save_original_input(self.linear_proj)
-
     def _checkpointed_attention_forward(
         self,
         query,
@@ -396,22 +368,16 @@ class Attention(MegatronModule, ABC):
             if rotary_pos_emb is not None:
                 q_pos_emb, k_pos_emb = rotary_pos_emb
                 key = inference_context.apply_rotary_emb_key(
-                    key, k_pos_emb, self.config, self.pg_collection.cp
+                    key, k_pos_emb, self.config, self.model_comm_pgs.cp
                 )
                 rotary_pos_emb = (q_pos_emb, None)  # key rotary emb has been applied
 
             # Append key/value data tensors to cache.
             inference_context.append_key_value_cache(self.layer_number, key, value)
 
-            _, max_seqlen_q = inference_context.cu_query_lengths()
-            if getattr(self.config, "cache_mla_latents", None) and max_seqlen_q > 1:
-                # Doing unabsorbed MLA Attention with cached mla latents (prefill/mixed mode)
-                kv_cache, _, block_table = inference_context.key_value_cache(self.layer_number)
-                # Uncompress the KV cache for prefill/mixed mode
-                key, value = self.uncompress_kv_from_cache(kv_cache)
-            else:
-                # Read key/value *pointer* tensors from cache.
-                key, value, block_table = inference_context.key_value_cache(self.layer_number)
+            # Read key/value *pointer* tensors from cache.
+            key, value, block_table = inference_context.key_value_cache(self.layer_number)
+
         return query, key, value, rotary_pos_emb, attn_mask_type, block_table
 
     @abstractmethod
@@ -477,6 +443,7 @@ class Attention(MegatronModule, ABC):
         cu_seqlens_q,
         cu_seqlens_k,
         seqlens_k,
+        seqlens_k_decode_only,
         block_table,
     ) -> Tensor:
         """Flash attention kernel for mixed decode and prefill samples.
@@ -490,6 +457,7 @@ class Attention(MegatronModule, ABC):
             cu_seqlens_q (Tensor): Cumulative query sequence lengths.
             cu_seqlens_k (Tensor): Cumulative key sequence lengths.
             seqlens_k (Tensor): key sequence lengths.
+            seqlens_k_decode_only (Tensor): key sequence lengths (decode_only).
             block_table (Tensor): KV cache chunk ids for all samples.
         Return:
             (Tensor) Attention output.
@@ -500,13 +468,10 @@ class Attention(MegatronModule, ABC):
         # Flash attn kernel.
         if max_seqlen_q > 1:
             q = q.squeeze(1)
-            if getattr(self, "softmax_scale", None) is not None:
-                softmax_scale = self.softmax_scale
-            else:
-                softmax_scale = q.shape[-1] ** -0.5
             if HAVE_FA3:
                 # TODO(ksanthanam): Replace with call to flash_attn_varlen_func once
                 # it accepts block_table
+                softmax_scale = q.shape[-1] ** -0.5
                 output_total, *unused = _flash_attn_forward(
                     q=q,
                     k=k,
@@ -551,53 +516,23 @@ class Attention(MegatronModule, ABC):
                     cu_seqlens_k,
                     max_seqlen_q,
                     max_seqlen_k,
-                    softmax_scale=softmax_scale,
                     causal=True,
                     block_table=block_table,
                 )
             output_total = output_total.unsqueeze(1)
         else:  # decode only
-            # If using MLA we use the FlashMLA kernel
-            if isinstance(self.config, MLATransformerConfig):
-                softmax_scale = self.softmax_scale
-
-                num_heads_k = 1  # Only a single head for MLA Flash
-                seq_len_q = 1  # Sequence length is 1 for decode
-                num_heads_q = self.num_attention_heads_per_partition
-                num_heads_per_head_k = seq_len_q * num_heads_q // num_heads_k
-
-                cache_seqlens = seqlens_k
-                tile_scheduler_metadata, num_splits = get_mla_metadata(
-                    cache_seqlens,  # cumulative key-lengths
-                    num_heads_per_head_k,  # decode-only lengths
-                    num_heads_k,  # per-head dim of V
-                )
-                head_dim_v = self.config.kv_lora_rank
-                kv_cache = k.unsqueeze(-2)
-                output_total, softmax_lse = flash_mla_with_kvcache(
-                    q,
-                    kv_cache,
-                    block_table,
-                    cache_seqlens,
-                    head_dim_v,
-                    tile_scheduler_metadata,
-                    num_splits,
-                    softmax_scale=softmax_scale,
-                    causal=True,
-                )
+            flash_attn_args = {
+                "q": q,
+                "k_cache": k,
+                "v_cache": v,
+                "cache_seqlens": seqlens_k_decode_only,
+                "causal": True,
+                "page_table" if HAVE_FA3 else "block_table": block_table,
+            }
+            if HAVE_FA3:
+                output_total = flash_attn3_with_kvcache(**flash_attn_args)
             else:
-                flash_attn_args = {
-                    "q": q,
-                    "k_cache": k,
-                    "v_cache": v,
-                    "cache_seqlens": seqlens_k,
-                    "causal": True,
-                    "page_table" if HAVE_FA3 else "block_table": block_table,
-                }
-                if HAVE_FA3:
-                    output_total = flash_attn3_with_kvcache(**flash_attn_args)
-                else:
-                    output_total = flash_attn_with_kvcache(**flash_attn_args)
+                output_total = flash_attn_with_kvcache(**flash_attn_args)
         return output_total
 
     def forward(
@@ -709,7 +644,6 @@ class Attention(MegatronModule, ABC):
         if (
             in_decode_mode
             and self.config.enable_cuda_graph
-            and self.config.cuda_graph_scope != "full_iteration"
             and inference_context.is_static_batching()
         ):
             raise ValueError(f"CUDA graphs must use flash decode with static batching!")
@@ -760,11 +694,11 @@ class Attention(MegatronModule, ABC):
                         q_pos_emb,
                         config=self.config,
                         cu_seqlens=cu_seqlens_q,
-                        cp_group=self.pg_collection.cp,
+                        cp_group=self.model_comm_pgs.cp,
                     )
                 else:
                     query = inference_context.apply_rotary_emb_query(
-                        query, q_pos_emb, self.config, cu_seqlens_q, self.pg_collection.cp
+                        query, q_pos_emb, self.config, cu_seqlens_q, self.model_comm_pgs.cp
                     )
             if k_pos_emb is not None:
                 key = apply_rotary_pos_emb(
@@ -772,7 +706,7 @@ class Attention(MegatronModule, ABC):
                     k_pos_emb,
                     config=self.config,
                     cu_seqlens=cu_seqlens_kv,
-                    cp_group=self.pg_collection.cp,
+                    cp_group=self.model_comm_pgs.cp,
                 )
 
             # TODO, can apply positional embedding to value_layer so it has
@@ -813,7 +747,9 @@ class Attention(MegatronModule, ABC):
                 # Dynamic batching attention kernel.
                 q, k, v = (query, key, value)
                 cu_query_lengths, max_seqlen_q = inference_context.cu_query_lengths()
-                cu_kv_lengths, kv_lengths, max_seqlen_k = inference_context.cu_kv_lengths()
+                cu_kv_lengths, kv_lengths, kv_lengths_decode_only, max_seqlen_k = (
+                    inference_context.cu_kv_lengths()
+                )
 
                 core_attn_out = self.flash_decode_and_prefill(
                     q,
@@ -824,6 +760,7 @@ class Attention(MegatronModule, ABC):
                     cu_query_lengths,
                     cu_kv_lengths,
                     kv_lengths,
+                    kv_lengths_decode_only,
                     block_table,
                 )
                 core_attn_out = rearrange(core_attn_out, 's b h d -> s b (h d)')
@@ -846,10 +783,6 @@ class Attention(MegatronModule, ABC):
 
         return output, bias
 
-    def set_for_recompute_input_layernorm(self):
-        """Set the attention layer for recompute input_layernorm. Only needed for fp8."""
-        raise NotImplementedError("set_for_recompute_input_layernorm is not implemented.")
-
 
 class SelfAttention(Attention):
     """Self-attention layer class
@@ -865,7 +798,7 @@ class SelfAttention(Attention):
         layer_number: int,
         attn_mask_type=AttnMaskType.padding,
         cp_comm_type: str = None,
-        pg_collection: ProcessGroupCollection = None,
+        model_comm_pgs: ModelCommProcessGroups = None,
     ):
         super().__init__(
             config=config,
@@ -874,7 +807,7 @@ class SelfAttention(Attention):
             attn_mask_type=attn_mask_type,
             attention_type="self",
             cp_comm_type=cp_comm_type,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )
 
         self.linear_qkv = build_module(
@@ -888,26 +821,48 @@ class SelfAttention(Attention):
             skip_bias_add=False,
             is_expert=False,
             tp_comm_buffer_name='qkv',
-            tp_group=self.pg_collection.tp,
+            tp_group=self.model_comm_pgs.tp,
         )
 
         if submodules.q_layernorm is not None:
-            self.q_layernorm = build_module(
-                submodules.q_layernorm,
-                hidden_size=self.hidden_size_per_attention_head,
-                config=self.config,
-                eps=self.config.layernorm_epsilon,
-            )
+            if not self.config.qk_layernorm_hidden_dim:
+                self.q_layernorm = build_module(
+                    submodules.q_layernorm,
+                    hidden_size=self.hidden_size_per_attention_head,
+                    config=self.config,
+                    eps=self.config.layernorm_epsilon,
+                )
+            else:
+                tp_world_size = get_tensor_model_parallel_world_size()
+                assert tp_world_size <= 1, "TP world size must be less than 1 for qk_layernorm_hidden_dim"
+                # nums_head_cur_rank = divide(self.config.num_attention_heads, tp_world_size)
+                self.q_layernorm = build_module(
+                    submodules.q_layernorm,
+                    hidden_size=self.query_projection_size,
+                    config=self.config,
+                    eps=self.config.layernorm_epsilon,
+                )
         else:
             self.q_layernorm = None
 
         if submodules.k_layernorm is not None:
-            self.k_layernorm = build_module(
-                submodules.k_layernorm,
-                hidden_size=self.hidden_size_per_attention_head,
-                config=self.config,
-                eps=self.config.layernorm_epsilon,
-            )
+            if not self.config.qk_layernorm_hidden_dim:
+                self.k_layernorm = build_module(
+                    submodules.k_layernorm,
+                    hidden_size=self.hidden_size_per_attention_head,
+                    config=self.config,
+                    eps=self.config.layernorm_epsilon,
+                )
+            else:
+                tp_world_size = get_tensor_model_parallel_world_size()
+                assert tp_world_size <= 1, "TP world size must be less than 1 for qk_layernorm_hidden_dim"
+                # nums_head_cur_rank = divide(self.config.num_attention_heads, tp_world_size)
+                self.k_layernorm = build_module(
+                    submodules.k_layernorm,
+                    hidden_size=self.kv_projection_size,
+                    config=self.config,
+                    eps=self.config.layernorm_epsilon,
+                )
         else:
             self.k_layernorm = None
 
@@ -1024,10 +979,24 @@ class SelfAttention(Attention):
         query = query.reshape(query.size(0), query.size(1), -1, self.hidden_size_per_attention_head)
 
         if self.q_layernorm is not None:
-            query = self.q_layernorm(query)
+            if not self.config.qk_layernorm_hidden_dim:
+                query = self.q_layernorm(query)
+            else:
+                # [sq, b, np, hn] -> [sq, b, 1, np * hn]
+                query_shape = list(query.shape)
+                query = query.reshape(query.size(0), query.size(1), 1, -1)
+                query = self.q_layernorm(query)
+                query = query.reshape(*query_shape)
 
         if self.k_layernorm is not None:
-            key = self.k_layernorm(key)
+            if not self.config.qk_layernorm_hidden_dim:
+                key = self.k_layernorm(key)
+            else:
+                # [sq, b, ng, hn] -> [sq, b, 1, ng * hn]
+                key_shape = list(key.shape)
+                key = key.reshape(key.size(0), key.size(1), 1, -1)
+                key = self.k_layernorm(key)
+                key = key.reshape(*key_shape)
 
         if self.config.test_mode:
             self.run_realtime_tests()
@@ -1047,12 +1016,6 @@ class SelfAttention(Attention):
         """Update weights for output projection layer"""
         self.linear_proj.backward_dw()
 
-    def set_for_recompute_input_layernorm(self):
-        """Set the attention layer for recompute input_layernorm. Only needed for fp8."""
-        from megatron.core.extensions.transformer_engine import set_save_original_input
-
-        set_save_original_input(self.linear_qkv)
-
 
 class CrossAttention(Attention):
     """Cross-attention layer class
@@ -1068,7 +1031,7 @@ class CrossAttention(Attention):
         layer_number: int,
         attn_mask_type=AttnMaskType.padding,
         cp_comm_type: str = None,
-        pg_collection: ProcessGroupCollection = None,
+        model_comm_pgs: ModelCommProcessGroups = None,
     ):
         super().__init__(
             config=config,
@@ -1077,7 +1040,7 @@ class CrossAttention(Attention):
             attn_mask_type=attn_mask_type,
             attention_type="cross",
             cp_comm_type=cp_comm_type,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )
 
         if self.config.num_query_groups != self.config.num_attention_heads:

