diff --git a/megatron/core/transformer/mlp.py b/megatron/core/transformer/mlp.py
index 7eafb2421..7da539614 100644
--- a/megatron/core/transformer/mlp.py
+++ b/megatron/core/transformer/mlp.py
@@ -1,6 +1,5 @@
 # Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
-import gc
-import logging
+
 import warnings
 from dataclasses import dataclass
 from typing import Optional, Union
@@ -15,11 +14,7 @@ from megatron.core.dist_checkpointing.mapping import (
     ShardedStateDict,
     ShardedTensorFactory,
 )
-from megatron.core.fusions.fused_bias_geglu import (
-    bias_geglu_impl,
-    quick_gelu,
-    weighted_bias_quick_geglu_impl,
-)
+from megatron.core.fusions.fused_bias_geglu import bias_geglu_impl
 from megatron.core.fusions.fused_bias_gelu import bias_gelu_impl
 from megatron.core.fusions.fused_bias_swiglu import bias_swiglu_impl, weighted_bias_swiglu_impl
 from megatron.core.transformer.module import MegatronModule
@@ -39,19 +34,10 @@ except ImportError:
     HAVE_TE = False
 
 
-logger = logging.getLogger(__name__)
-
-
 # pylint: disable=missing-class-docstring
 @dataclass
 class MLPSubmodules:
-    """
-    The dataclass for ModuleSpecs of MLP submodules
-    including  linear fc1, activation function, linear fc2.
-    """
-
     linear_fc1: Union[ModuleSpec, type] = None
-    activation_func: Union[ModuleSpec, type] = None
     linear_fc2: Union[ModuleSpec, type] = None
 
 
@@ -118,10 +104,7 @@ class MLP(MegatronModule):
             tp_group=tp_group,
         )
 
-        if self.config.use_te_activation_func and not (submodules.activation_func is None):
-            self.activation_func = build_module(submodules.activation_func, config=self.config)
-        else:
-            self.activation_func = self.config.activation_func
+        self.activation_func = self.config.activation_func
 
         self.linear_fc2 = build_module(
             submodules.linear_fc2,
@@ -145,15 +128,7 @@ class MLP(MegatronModule):
         nvtx_range_pop(suffix="linear_fc1")
 
         nvtx_range_push(suffix="activation")
-        if self.config.use_te_activation_func:
-            if bias_parallel is not None:
-                intermediate_parallel = intermediate_parallel + bias_parallel
-            intermediate_parallel = self.activation_func(intermediate_parallel)
-            if per_token_scale is not None:
-                original_dtype = intermediate_parallel.dtype
-                intermediate_parallel = intermediate_parallel * per_token_scale.unsqueeze(-1)
-                intermediate_parallel = intermediate_parallel.to(original_dtype)
-        elif self.config.bias_activation_fusion:
+        if self.config.bias_activation_fusion:
             if per_token_scale is not None:
                 if self.activation_func == F.silu and self.config.gated_linear_unit:
                     # dtype is handled inside the fused kernel
@@ -163,19 +138,8 @@ class MLP(MegatronModule):
                         per_token_scale.unsqueeze(-1),
                         self.config.activation_func_fp8_input_store,
                     )
-                elif self.activation_func == quick_gelu and self.config.gated_linear_unit:
-                    intermediate_parallel = weighted_bias_quick_geglu_impl(
-                        intermediate_parallel,
-                        bias_parallel,
-                        per_token_scale.unsqueeze(-1),
-                        self.config.activation_func_fp8_input_store,
-                        self.config.glu_linear_offset,
-                        self.config.activation_func_clamp_value,
-                    )
                 else:
-                    raise ValueError(
-                        "Only support fusion of swiglu and quick_gelu with per_token_scale in MLP."
-                    )
+                    raise ValueError("Only support fusion of swiglu with per_token_scale in MLP.")
             else:
                 if self.activation_func == F.gelu:
                     if self.config.gated_linear_unit:
@@ -202,13 +166,8 @@ class MLP(MegatronModule):
             if self.config.gated_linear_unit:
 
                 def glu(x):
-                    x_glu, x_linear = torch.chunk(x, 2, dim=-1)
-                    if (val := self.config.activation_func_clamp_value) is not None:
-                        x_glu = x_glu.clamp(min=None, max=val)
-                        x_linear = x_linear.clamp(min=-val, max=val)
-                    return self.config.activation_func(x_glu) * (
-                        x_linear + self.config.glu_linear_offset
-                    )
+                    x = torch.chunk(x, 2, dim=-1)
+                    return self.config.activation_func(x[0]) * x[1]
 
                 intermediate_parallel = glu(intermediate_parallel)
             else:
@@ -234,29 +193,19 @@ class MLP(MegatronModule):
     def sharded_state_dict(
         self, prefix: str = "", sharded_offsets: tuple = (), metadata: Optional[dict] = None
     ) -> ShardedStateDict:
-        """Return the sharded state dictionary of the module."""
         sharded_state_dict = {}
-        singleton_local_shards = (metadata or {}).get('singleton_local_shards', False)
         for name, module in self._modules.items():
             sub_sd = module.sharded_state_dict(f"{prefix}{name}.", sharded_offsets, metadata)
             if self.config.gated_linear_unit and name == "linear_fc1":
                 for k, v in sub_sd.items():
                     if k in (f"{prefix}{name}.weight", f"{prefix}{name}.bias"):
-                        sub_sd[k] = apply_swiglu_sharded_factory(
-                            v, sharded_offsets, singleton_local_shards
-                        )
+                        sub_sd[k] = apply_swiglu_sharded_factory(v, sharded_offsets)
             sharded_state_dict.update(sub_sd)
         return sharded_state_dict
 
-    def backward_dw(self):
-        self.linear_fc2.backward_dw()
-        self.linear_fc1.backward_dw()
-
 
 # pylint: disable=missing-function-docstring
-def apply_swiglu_sharded_factory(
-    original_sh_ten, sharded_offsets, singleton_local_shards: bool = False
-):
+def apply_swiglu_sharded_factory(original_sh_ten, sharded_offsets):
     # We must split the tensor into 2 parts, each sharded separately.
     # This requires a ShardedTensorFactory which `chunk`s during saving
     # and `cat`s during loading
@@ -278,25 +227,13 @@ def apply_swiglu_sharded_factory(
     def sh_ten_build_fn(
         key: str, t: torch.Tensor, replica_id: ReplicaId, flattened_range: Optional[slice]
     ):
-        if singleton_local_shards:
-            offset_w = (swiglu_shard_axis + prepend_axis_num, rank_offset, axis_frag)
-            offset_v = (swiglu_shard_axis + prepend_axis_num, rank_offset, axis_frag)
-            w_key = f'{key}_w'
-            v_key = f'{key}_v'
-        else:
-            offset_w = (swiglu_shard_axis + prepend_axis_num, rank_offset, axis_frag * 2)
-            offset_v = (
-                swiglu_shard_axis + prepend_axis_num,
-                rank_offset + axis_frag,
-                axis_frag * 2,
-            )
-            w_key = key
-            v_key = key
+        offset_w = (swiglu_shard_axis + prepend_axis_num, rank_offset, axis_frag * 2)
+        offset_v = (swiglu_shard_axis + prepend_axis_num, rank_offset + axis_frag, axis_frag * 2)
         if flattened_range is None:
             tensor_w, tensor_v = torch.chunk(t, 2, dim=swiglu_shard_axis)
             return [
                 ShardedTensor.from_rank_offsets(
-                    w_key,
+                    key,
                     tensor_w,
                     *sharded_offsets,
                     offset_w,
@@ -304,7 +241,7 @@ def apply_swiglu_sharded_factory(
                     prepend_axis_num=prepend_axis_num,
                 ),
                 ShardedTensor.from_rank_offsets(
-                    v_key,
+                    key,
                     tensor_v,
                     *sharded_offsets,
                     offset_v,
@@ -313,10 +250,6 @@ def apply_swiglu_sharded_factory(
                 ),
             ]
         else:
-            if singleton_local_shards:
-                raise NotImplementedError(
-                    'singleton_local_shards not implemented for SwiGLU MLP flattened tensors'
-                )
             # Here we need to map a slice `t` (`flattened_range` specifies slice start and stop)
             # of the *original* flattened tensor into slices `w` and `v` of chunked
             # and flattened tensor.
@@ -378,17 +311,7 @@ def apply_swiglu_sharded_factory(
 
     def sh_ten_merge_fn(sub_state_dict):
         with torch.no_grad():
-            try:
-                return torch.cat(sub_state_dict)
-            except (RuntimeError, torch.cuda.OutOfMemoryError) as e:
-                logger.warning(
-                    f"CUDA OutOfMemoryError encountered during tensors merging."
-                    f" Switching to CPU merge. (Error: {e})"
-                )
-                merged_sub_state_dict = torch.cat([t.cpu() for t in sub_state_dict])
-                gc.collect()
-                torch.cuda.empty_cache()
-                return merged_sub_state_dict
+            return torch.cat(sub_state_dict)
 
     return ShardedTensorFactory(
         original_sh_ten.key,

