diff --git a/megatron/core/transformer/transformer_layer.py b/megatron/core/transformer/transformer_layer.py
index 42d077e2e..6e9251447 100644
--- a/megatron/core/transformer/transformer_layer.py
+++ b/megatron/core/transformer/transformer_layer.py
@@ -14,8 +14,8 @@ from megatron.core import parallel_state, tensor_parallel
 from megatron.core.dist_checkpointing.mapping import ShardedStateDict
 from megatron.core.dist_checkpointing.utils import apply_prefix_mapping
 from megatron.core.packed_seq_params import PackedSeqParams
-from megatron.core.process_groups_config import ProcessGroupCollection
-from megatron.core.transformer.cuda_graphs import CudaGraphManager, is_graph_capturing
+from megatron.core.process_groups_config import ModelCommProcessGroups
+from megatron.core.transformer.cuda_graphs import CudaGraphManager
 from megatron.core.transformer.enums import LayerType
 from megatron.core.transformer.identity_op import IdentityFuncOp, IdentityOp
 from megatron.core.transformer.mlp import MLP
@@ -24,7 +24,6 @@ from megatron.core.transformer.spec_utils import ModuleSpec, build_module
 from megatron.core.transformer.transformer_config import TransformerConfig
 from megatron.core.utils import (
     deprecate_inference_params,
-    get_pg_rank,
     is_te_min_version,
     log_single_rank,
     make_viewless_tensor,
@@ -35,18 +34,18 @@ from megatron.core.utils import (
 logger = logging.getLogger(__name__)
 
 
-def get_transformer_layer_offset(
-    config: TransformerConfig, vp_stage: Optional[int] = None, pp_rank: Optional[int] = None
-):
+def get_transformer_layer_offset(config: TransformerConfig, vp_stage: Optional[int] = None, is_dualpipev_first_chunk: Optional[bool] = False):
     """Get the index offset of current pipeline stage, given the level of pipelining."""
-    if pp_rank is None:
-        pp_rank = parallel_state.get_pipeline_model_parallel_rank()
-
-    is_first_pp_stage = pp_rank == 0
+    pipeline_rank = parallel_state.get_pipeline_model_parallel_rank()
+    if not parallel_state.is_inside_encoder():
+        pp_decoder_start = parallel_state.get_pipeline_model_parallel_decoder_start()
+        if pp_decoder_start is not None:
+            pipeline_rank = pipeline_rank - pp_decoder_start
 
     if config.pipeline_model_parallel_size > 1:
-
-        if config.pipeline_model_parallel_layout:
+        if config.enable_hetero:
+            offset = sum(([0] + config.hetero_pipeline_layer_split)[: pipeline_rank + 1])
+        elif config.pipeline_model_parallel_layout:
             offset = config.pipeline_model_parallel_layout.get_layer_offset(
                 layer_type=LayerType.decoder, vp_stage=vp_stage
             )
@@ -88,10 +87,6 @@ def get_transformer_layer_offset(
                 - num_layers_in_last_pipeline_stage
             )
 
-            middle_pipeline_rank = (
-                pp_rank if config.num_layers_in_first_pipeline_stage is None else pp_rank - 1
-            )
-
             if (vp_size := config.virtual_pipeline_model_parallel_size) is not None:
                 assert (
                     vp_stage is not None
@@ -125,13 +120,13 @@ def get_transformer_layer_offset(
                 )
 
                 # Calculate the layer offset with interleaved uneven pipeline parallelism
-                if pp_rank == 0:
+                if pipeline_rank == 0:
                     offset = vp_stage * total_virtual_chunks
                 else:
                     offset = (
                         vp_stage * total_virtual_chunks
                         + num_layers_per_virtual_model_chunk_in_first_pipeline_stage
-                        + middle_pipeline_rank
+                        + (pipeline_rank - 1)
                         * (
                             num_layers_per_vritual_model_chunk_in_middle_pipeline_stage
                             // middle_pipeline_stages
@@ -143,12 +138,74 @@ def get_transformer_layer_offset(
                 else:
                     num_layers_per_pipeline_rank = 0
 
-                if pp_rank == 0:
-                    offset = 0
+                middle_pipeline_rank = (
+                    pipeline_rank
+                    if config.num_layers_in_first_pipeline_stage is None
+                    else pipeline_rank - 1
+                )
+
+                if not config.use_dualpipev:
+                    if pipeline_rank == 0:
+                        offset = 0
+                    else:
+                        offset = (
+                            middle_pipeline_rank * num_layers_per_pipeline_rank
+                        ) + num_layers_in_first_pipeline_stage
+                ######### FlagScale Begin ########
                 else:
-                    offset = (
-                        middle_pipeline_rank * num_layers_per_pipeline_rank
-                    ) + num_layers_in_first_pipeline_stage
+                    # for dualpipev
+                    num_layers_in_first_pipeline_stage_first_chunk = num_layers_in_first_pipeline_stage // 2
+                    if num_layers_in_first_pipeline_stage % 2 != 0:
+                        num_layers_in_first_pipeline_stage_first_chunk = num_layers_in_first_pipeline_stage_first_chunk + 1
+                    num_layers_in_first_pipeline_stage_second_chunk = num_layers_in_first_pipeline_stage - num_layers_in_first_pipeline_stage_first_chunk
+
+                    num_layers_in_last_pipeline_stage_first_chunk = num_layers_in_last_pipeline_stage // 2
+                    if num_layers_in_last_pipeline_stage % 2 != 0:
+                        num_layers_in_last_pipeline_stage_first_chunk = num_layers_in_last_pipeline_stage_first_chunk + 1
+                    num_layers_in_last_pipeline_stage_second_chunk = num_layers_in_last_pipeline_stage - num_layers_in_last_pipeline_stage_first_chunk
+
+                    num_layers_per_pipeline_rank_first_chunk = num_layers_per_pipeline_rank // 2
+                    if num_layers_per_pipeline_rank % 2 != 0:
+                        num_layers_per_pipeline_rank_first_chunk = num_layers_per_pipeline_rank_first_chunk + 1
+                    num_layers_per_pipeline_rank_second_chunk = num_layers_per_pipeline_rank - num_layers_per_pipeline_rank_first_chunk
+
+                    # process first chunk
+                    if is_dualpipev_first_chunk:
+                        middle_pipeline_rank = (
+                            pipeline_rank
+                            if config.num_layers_in_first_pipeline_stage is None
+                            else pipeline_rank - 1
+                        )
+                        if pipeline_rank == 0:
+                            offset = 0
+                        else:
+                            offset = (
+                                middle_pipeline_rank * num_layers_per_pipeline_rank_first_chunk
+                            ) + num_layers_in_first_pipeline_stage_first_chunk
+
+                    # process second chunk
+                    else:
+                        start_offset = (
+                            config.pipeline_model_parallel_size * num_layers_per_pipeline_rank_first_chunk
+                        )
+                        if num_layers_in_first_pipeline_stage_first_chunk > 0:
+                            start_offset = start_offset - num_layers_per_pipeline_rank_first_chunk + num_layers_in_first_pipeline_stage_first_chunk
+                        if num_layers_in_last_pipeline_stage_first_chunk > 0:
+                            start_offset = start_offset - num_layers_per_pipeline_rank_first_chunk + num_layers_in_last_pipeline_stage_first_chunk
+
+                        middle_pipeline_rank = (
+                            config.pipeline_model_parallel_size - 1 - pipeline_rank
+                            if config.num_layers_in_last_pipeline_stage is None
+                            else config.pipeline_model_parallel_size - 1 - pipeline_rank - 1
+                        )
+
+                        if pipeline_rank == config.pipeline_model_parallel_size - 1:
+                            offset = start_offset
+                        else:
+                            offset = start_offset + (
+                                middle_pipeline_rank * num_layers_per_pipeline_rank_second_chunk
+                            ) + num_layers_in_last_pipeline_stage_second_chunk
+                ######### FlagScale End ########
         else:
             num_layers = config.num_layers
 
@@ -162,9 +219,6 @@ def get_transformer_layer_offset(
 
             num_layers_per_pipeline_rank = num_layers // config.pipeline_model_parallel_size
 
-            # import here to avoid circular import
-            from megatron.core.pipeline_parallel.utils import is_vp_first_stage
-
             if (vp_size := config.virtual_pipeline_model_parallel_size) is not None:
                 assert (
                     vp_stage is not None
@@ -172,21 +226,43 @@ def get_transformer_layer_offset(
 
                 num_layers_per_virtual_rank = num_layers_per_pipeline_rank // vp_size
                 total_virtual_chunks = num_layers // vp_size
-                offset = vp_stage * total_virtual_chunks + (pp_rank * num_layers_per_virtual_rank)
+                offset = vp_stage * total_virtual_chunks + (
+                    pipeline_rank * num_layers_per_virtual_rank
+                )
 
                 # Reduce the offset of embedding layer from the total layer number
-                if config.account_for_embedding_in_pipeline_split and not (
-                    is_vp_first_stage(vp_stage, vp_size) and is_first_pp_stage
+                if (
+                    config.account_for_embedding_in_pipeline_split
+                    and not parallel_state.is_pipeline_first_stage(
+                        ignore_virtual=False, vp_stage=vp_stage
+                    )
                 ):
                     offset -= 1
             else:
-                offset = pp_rank * num_layers_per_pipeline_rank
+                if not config.use_dualpipev:
+                    offset = pipeline_rank * num_layers_per_pipeline_rank
+
+                    # Reduce the offset of embedding layer from the total layer number
+                    if (
+                        config.account_for_embedding_in_pipeline_split
+                        and not parallel_state.is_pipeline_first_stage(
+                            ignore_virtual=False, vp_stage=vp_stage
+                        )
+                    ):
+                        offset -= 1
+                ######### FlagScale Begin ########
+                else:
+                    num_layers_per_pipeline_rank_first_chunk = num_layers_per_pipeline_rank // 2
+                    if num_layers_per_pipeline_rank % 2 != 0:
+                        num_layers_per_pipeline_rank_first_chunk = num_layers_per_pipeline_rank_first_chunk + 1
+                    num_layers_per_pipeline_rank_second_chunk = num_layers_per_pipeline_rank - num_layers_per_pipeline_rank_first_chunk
+
+                    if is_dualpipev_first_chunk:
+                        offset = pipeline_rank * num_layers_per_pipeline_rank_first_chunk
+                    else:
+                        offset = config.num_layers - ((pipeline_rank+1) * num_layers_per_pipeline_rank_second_chunk)
+                ######### FlagScale End ########
 
-                # Reduce the offset of embedding layer from the total layer number
-                if config.account_for_embedding_in_pipeline_split and not (
-                    is_vp_first_stage(vp_stage, vp_size) and is_first_pp_stage
-                ):
-                    offset -= 1
     else:
         offset = 0
     return offset
@@ -265,25 +341,23 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
         submodules: TransformerLayerSubmodules,
         layer_number: int = 1,
         hidden_dropout: Optional[float] = None,
-        pg_collection: Optional[ProcessGroupCollection] = None,
+        model_comm_pgs: Optional[ModelCommProcessGroups] = None,
         vp_stage: Optional[int] = None,
     ):
         super().__init__(config=config)
 
         # Enable cuda graphs.
-        if (
-            config.enable_cuda_graph and config.cuda_graph_scope != "full_iteration"
-        ) or config.external_cuda_graph:
+        if config.enable_cuda_graph or config.external_cuda_graph:
             assert not (
                 config.enable_cuda_graph and config.external_cuda_graph
             ), "Cudagraphs and external cudagraphs cannot be enabled at the same time"
-            if config.enable_cuda_graph and config.cuda_graph_scope != "full_iteration":
+            if config.enable_cuda_graph:
                 if not self.training:
                     # Cudagraphs for inference are only enabled with the flash decoding kernel
                     assert (
                         self.config.flash_decode
                     ), "--flash-decode is required to use CUDA graphs during inference"
-                self.cudagraph_manager = CudaGraphManager(config, vp_stage=vp_stage)
+                self.cudagraph_manager = CudaGraphManager(config)
             else:
                 # List to store CUDA graphs. A list of `N` CUDA graphs for this layer where N is
                 # the number of microbatches. Multiple CUDA graphs per layer is required to support
@@ -299,14 +373,11 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
                 self.cuda_graph_manual_hooks = []
                 self.current_microbatch = -1
 
-        if pg_collection is None:
-            pg_collection = ProcessGroupCollection.use_mpu_process_groups()
-        self.pg_collection = pg_collection
+        if model_comm_pgs is None:
+            model_comm_pgs = ModelCommProcessGroups.use_mpu_process_groups()
 
         self.submodules_config = submodules
-        self.layer_number = layer_number + get_transformer_layer_offset(
-            self.config, vp_stage, get_pg_rank(pg_collection.pp)
-        )
+        self.layer_number = layer_number + get_transformer_layer_offset(self.config, vp_stage)
         self.hidden_dropout = config.hidden_dropout if hidden_dropout is None else hidden_dropout
 
         # [Module 1: Input Layernorm] Optional Layernorm on the input data
@@ -325,7 +396,7 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
             else:
                 attention_optional_kwargs["cp_comm_type"] = config.cp_comm_type
 
-        attention_optional_kwargs["pg_collection"] = pg_collection
+        attention_optional_kwargs["model_comm_pgs"] = model_comm_pgs
 
         # [Module 2: SelfAttention]
         self.self_attention = build_module(
@@ -367,27 +438,21 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
         # [Module 8: MLP block]
         additional_mlp_kwargs = {}
         # import here to avoid circular import
-        from megatron.core.extensions.transformer_engine import TEFusedMLP
         from megatron.core.transformer.moe.experts import GroupedMLP, SequentialMLP, TEGroupedMLP
         from megatron.core.transformer.moe.moe_layer import MoELayer
 
-        # MLP expects tp_group but MoELayer expects pg_collection to be passed in.
-        # We can change MLP to accept pg_collection but it makes the logic implicit
+        # MLP expects tp_group but MoELayer expects model_comm_pgs to be passed in.
+        # We can change MLP to accept model_comm_pgs but it makes the logic implicit
         # The conditional below is to make the logic explicit
         # if submodules.mlp is not a ModuleSpec,we dont have to handle passing additional kwargs
         if isinstance(submodules.mlp, ModuleSpec):
             if submodules.mlp.module in (MoELayer, GroupedMLP, TEGroupedMLP, SequentialMLP):
-                additional_mlp_kwargs["pg_collection"] = pg_collection
+                additional_mlp_kwargs["model_comm_pgs"] = model_comm_pgs
             elif submodules.mlp.module == MLP:
                 assert hasattr(
-                    pg_collection, 'tp'
+                    model_comm_pgs, 'tp'
                 ), 'TP process group is required for MLP in TransformerLayer'
-                additional_mlp_kwargs["tp_group"] = pg_collection.tp
-            elif TEFusedMLP is not None and submodules.mlp.module == TEFusedMLP:
-                assert hasattr(
-                    pg_collection, 'tp'
-                ), 'TP process group is required for TEFusedMLP in TransformerLayer'
-                additional_mlp_kwargs["tp_group"] = pg_collection.tp
+                additional_mlp_kwargs["tp_group"] = model_comm_pgs.tp
             else:
                 log_single_rank(
                     logger,
@@ -406,25 +471,12 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
         self.recompute_mlp = False
         if self.config.recompute_granularity == 'selective':
             if "layernorm" in self.config.recompute_modules:
-                if (
-                    not isinstance(self.input_layernorm, IdentityOp)
-                    and not self.config.external_cuda_graph
-                ):
+                if not isinstance(self.input_layernorm, IdentityOp):
                     self.recompute_input_layernorm = True
-                    if self.config.fp8:
-                        self.self_attention.set_for_recompute_input_layernorm()
                 if not isinstance(self.pre_mlp_layernorm, IdentityOp):
                     self.recompute_pre_mlp_layernorm = True
-                    if self.config.fp8:
-                        if isinstance(self.mlp, MoELayer):
-                            self.mlp.set_for_recompute_pre_mlp_layernorm()
-                        else:
-                            from megatron.core.extensions.transformer_engine import (
-                                set_save_original_input,
-                            )
-
-                            set_save_original_input(self.mlp.linear_fc1)
             if "mlp" in self.config.recompute_modules:
+
                 if not isinstance(self.mlp, MoELayer):
                     self.recompute_mlp = True
 
@@ -457,10 +509,6 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
         This method calls the core computation of a transformer layer, including
         self-attention, cross-attention (if applicable), and feed-forward operations.
         """
-        # Remove 'dynamic_inference_decode_only' from kwargs if present
-        # this is only used to uniquely identify decode and non-decode cuda graph
-        # runners in the cuda graph manager
-        kwargs.pop("dynamic_inference_decode_only", None)
         hidden_states, context = self._forward_attention(*args, **kwargs)
         output = self._forward_mlp(hidden_states, kwargs.get("inference_context", None))
         return output, context
@@ -617,7 +665,7 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
                     self.mlp,
                     False,
                     tensor_parallel.random.get_cuda_rng_tracker,
-                    self.pg_collection.tp,
+                    parallel_state.get_tensor_model_parallel_group(),
                     pre_mlp_layernorm_output,
                 )
             else:
@@ -745,6 +793,7 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
                 self.self_attention,
                 self.pre_cross_attn_layernorm,
                 self.cross_attention,
+                self.pre_mlp_layernorm,
             ]
 
         param_modules = []
@@ -815,8 +864,7 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
                         device=torch.cuda.current_device(),
                     )
 
-                if not is_te_min_version("1.10.0"):
-                    # TE version < 1.10.0 does not support keyword arguments with CUDA graph.
+                if not is_te_min_version("1.10.0", check_equality=False):
                     for k, v in kwargs.items():
                         if k == "attention_mask":
                             if v is not None:
@@ -851,7 +899,7 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
             hook(*hook_args)
         cuda_graph_output = self.cuda_graphs[cg_index](*cudagraph_args, **cudagraph_kwargs)
 
-        if cudagraph_kwargs.get('context') is not None:
+        if cudagraph_kwargs['context'] is not None:
             context = cuda_graph_output[-1]
             cuda_graph_output = cuda_graph_output[:-1]
         else:
@@ -867,46 +915,27 @@ class TransformerLayer(MegatronModule, BaseTransformerLayer):
     def __call__(self, *args, **kwargs):
         # Training and validation mode CUDA graphs
         if hasattr(self, 'cudagraph_manager') and kwargs.get('inference_context') is None:
-            # Set the is_first_microbatch flag for weight caching
-            current_microbatch = getattr(self, 'current_microbatch', 0)
-            self.cudagraph_manager.set_is_first_microbatch(current_microbatch == 0)
             return self.cudagraph_manager(self, args, kwargs)
         # Inference mode. CUDA graphs are used in the decode phase only, when attn mask is None
         elif not self.training and (
             hasattr(self, 'cudagraph_manager')
             and kwargs['attention_mask'] is None
             and (
-                (kwargs.get('inference_context') is not None)
-                or (kwargs.get('inference_params') is not None)
+                (
+                    kwargs.get('inference_context') is not None
+                    and kwargs['inference_context'].is_decode_only()
+                )
+                or (
+                    kwargs.get('inference_params') is not None
+                    and kwargs['inference_params'].is_decode_only()
+                )
             )
         ):
             assert (
                 kwargs.get('attention_mask') is None
-            ), f"Attention mask must not be set when using CUDA graphs with inference."
-
-            # it can happen that non-decode steps have a token count greater than the max
-            # supported cuda graph token count. In that case this flag will be set to
-            # False by initialize_attention, and we should not use cuda graphs.
-            if kwargs['inference_context'].is_static_batching():
-                using_cuda_graph = kwargs['inference_context'].is_decode_only()
-            else:
-                using_cuda_graph = kwargs['inference_context'].using_cuda_graph_this_step()
-            if using_cuda_graph:
-                # dynamic_inference_decode_only is not a real argument to forward, it is only used
-                # to differentiate the cuda graph used for decode from the one used for non-decode
-                # inference.
-                kwargs["dynamic_inference_decode_only"] = kwargs[
-                    'inference_context'
-                ].is_decode_only()
-                # Set the is_first_microbatch flag for weight caching
-                current_microbatch = getattr(self, 'current_microbatch', 0)
-                self.cudagraph_manager.set_is_first_microbatch(current_microbatch == 0)
-                return self.cudagraph_manager(self, args, kwargs)
-        elif (
-            self.config.external_cuda_graph
-            and self.training
-            and (is_graph_capturing() or self.cuda_graphs)
-        ):
+            ), f"Attention mask must not be set when using CUDA graphs for decode"
+            return self.cudagraph_manager(self, args, kwargs)
+        elif self.config.external_cuda_graph and self.training:
             if not self.cuda_graphs:
                 # Do CUDA Graphs capture.
                 cuda_graph_func = self._cuda_graph_capture

