diff --git a/megatron/core/transformer/transformer_config.py b/megatron/core/transformer/transformer_config.py
index 5ff62f74c..27b1b87b8 100644
--- a/megatron/core/transformer/transformer_config.py
+++ b/megatron/core/transformer/transformer_config.py
@@ -2,7 +2,7 @@
 
 import warnings
 from dataclasses import dataclass
-from typing import Callable, List, Literal, Optional, Tuple, Union
+from typing import Callable, List, Optional, Tuple, Union
 
 import torch
 import torch.nn.functional as F
@@ -12,7 +12,6 @@ from megatron.core.quantization.quant_config import RecipeConfig
 from megatron.core.transformer.enums import AttnBackend
 from megatron.core.transformer.pipeline_parallel_layer_layout import PipelineParallelLayerLayout
 
-from ..fusions.fused_bias_geglu import quick_gelu
 from ..model_parallel_config import ModelParallelConfig
 from ..utils import (
     get_te_version,
@@ -108,11 +107,6 @@ class TransformerConfig(ModelParallelConfig):
     softmax_scale: Optional[float] = None
     """Softmax scale for attention scaling."""
 
-    softmax_type: Literal['vanilla', 'off-by-one', 'learnable'] = 'vanilla'
-    """Applies modified softmax from https://www.evanmiller.org/attention-is-off-by-one.html. 
-       Supports both TE FusedAttention and local unfused attention. Supports both a fixed offset and 
-       and learnable offset."""
-
     num_query_groups: Optional[int] = None
     """Number of query groups for group query attention. If None, normal attention is used."""
 
@@ -161,14 +155,6 @@ class TransformerConfig(ModelParallelConfig):
     """Store the input of MLP activation function in FP8 for backprop to save memory.
     The stored input is casted back to the original precision before backprop compuatation."""
 
-    glu_linear_offset: float = 0.0
-    """Offset term in the GLU activation function: activation_func(x[0]) * (x[1] + offset). Only 
-    used when gated_linear_unit is True"""
-
-    activation_func_clamp_value: Optional[float] = None
-    """Clamp the output of the linear_fc1 in the activation function. Only used when activation_func
-    is quick_gelu."""
-
     num_moe_experts: Optional[int] = None
     """Number of experts to use for MoE layer. When set, it replaces MLP with MoE layer. Set to None
     for no MoE."""
@@ -181,11 +167,6 @@ class TransformerConfig(ModelParallelConfig):
     """If not None, then will use sliding window attention. The size of the window is specified by
     the numbers inside the tuple; -1 is special value meaning "infinite window size"."""
 
-    window_attn_skip_freq: Optional[Union[int, List[int]]] = None
-    """Frequency of full attention layers among sliding window attention layers. Accepts either:
-    - An integer N: Represents a (N-1):1 ratio, one full attention layer after (N-1) SWA layers.
-    - A list that defines a custom pattern, e.g.: [1,1,1,1,0,0,0,0], where 1 represents SWA. """
-
     normalization: str = "LayerNorm"
     """Which norm to use for normalization layers, valid options are `LayerNorm` and `RMSNorm`."""
 
@@ -230,22 +211,10 @@ class TransformerConfig(ModelParallelConfig):
     """Standard deviation of the zero mean normal for the default initialization method, not used if
     init_method and output_layer_init_method are provided."""
 
-    embedding_init_method: Optional[Callable] = None
-    """
-    Method to initialize weights of the embedding layer. If None, will be set as described 
-    in init_method above.
-    """
-
-    embedding_init_method_std: Optional[float] = None
-    """
-    Standard deviation of the zero mean normal for the default initialization method for the 
-    embedding layer. If None, will be set to init_method_std.
-    """
-
     init_model_with_meta_device: bool = False
     """
     If True, initializes the model with the meta device. This is helpful for
-    training of very large models. This feature is only works when megatron fsdp is turned on.
+    training of very large models. This feature is only works when custom fsdp is turned on.
     """
 
     ####################
@@ -286,9 +255,6 @@ class TransformerConfig(ModelParallelConfig):
     apply_rope_fusion: bool = False
     """If True, use fused RoPE kernel."""
 
-    use_fused_weighted_squared_relu: bool = False
-    """If True, uses fused weighted squared relu kernel when using MoE."""
-
     ####################
     # activation recomputation
     ####################
@@ -317,12 +283,21 @@ class TransformerConfig(ModelParallelConfig):
     the number of transformer layers to recompute within each pipeline stage.  Must be None for
     'selective' activation checkpointing."""
 
+    recompute_granularity_per_stage_micro_batch: list = None
+    """Same as recompute_granularity but for each stage and each micro-batch."""
+
+    recompute_method_per_stage_micro_batch: list = None
+    """Same as recompute_method but for each stage and each micro-batch."""
+
+    recompute_num_layers_per_stage_micro_batch: list = None
+    """Same as recompute_num_layers but for each stage and each micro-batch."""
+
     distribute_saved_activations: Optional[bool] = None
     """If True, distribute recomputed activations across the model parallel group."""
 
     recompute_modules: Optional[List[str]] = None
     """The submodules to recompute.
-    choices: "core_attn", "moe_act", "layernorm", "mla_up_proj", "mlp", "moe", "shared_experts".
+    choices: "core_attn", "moe_act", "layernorm", "mla_up_proj", "mlp", "moe".
     default: ["core_attn"].
     "core_attn": recompute the core attention part of the transformer layer.
     "moe_act": recompute the MoE MLP activation function.
@@ -330,9 +305,8 @@ class TransformerConfig(ModelParallelConfig):
     "mla_up_proj": recompute the MLA up projection and RoPE applying parts.
     "mlp": recompute the dense MLP submodule.
     "moe": recompute the MoE layer.
-    "shared_experts": recompute the shared experts in the MoE layer.
     "moe_act", "layernorm", and "mla_up_proj" use output-discarding checkpointing,
-    "core_attn", "mlp", "moe", and "shared_experts" use normal checkpointing.
+    "core_attn", "mlp", and "moe" uses normal checkpointing.
     """
 
     ####################
@@ -402,20 +376,10 @@ class TransformerConfig(ModelParallelConfig):
     """Use the kitchen extension for transformer quantization."""
 
     ####################
-    # fp4 related
+    # DualPipeV related
     ####################
-    fp4: Optional[str] = None
-    """If set, enables the use of FP4 precision through Transformer Engine. Currently only 
-    supports 'nvfp4' which uses NVFP4BlockScaling recipe (requires TE >= 2.7.0.dev0)."""
-
-    fp4_recipe: Optional[str] = "nvfp4"
-    """If set, enables the use of FP4 precision through Transformer Engine. Currently only
-    'nvfp4' is supported which uses NVFP4BlockScaling recipe for Blackwell+ architecture."""
-
-    fp4_param: bool = False
-    """If set, keep the parameters in fp4 precision to save memory. This option must be used
-    together with fp4 mode (i.e., TransformerConfig.fp4 is not None). Note that not all parameters
-    will be converted to fp4; for example, biases will remain unchanged."""
+    use_dualpipev: bool = False
+    moe_fb_overlap: bool = False
 
     ####################
     # MoE related
@@ -438,19 +402,12 @@ class TransformerConfig(ModelParallelConfig):
     moe_ffn_hidden_size: Optional[int] = None
     """MoE Feed-Forward Network hidden size"""
 
-    moe_router_load_balancing_type: Union[str, List[str]] = "aux_loss"
-    """The load balancing strategy for the router.
-    Options:
-    - "aux_loss": Load balancing loss used in GShard and SwitchTransformer, calculated at
-    micro-batch level.
-    - "seq_aux_loss": Load balancing loss used in DeepSeekV2 and DeepSeekV3, computes loss
-    for each individual sample.
-    - "global_aux_loss": Load balancing loss calculated at global batch level.
-    - "sinkhorn": Balancing algorithm used in S-BASE.
-    - "none": No load balancing.
-    A list of strings can be provided to combine multiple aux-loss load balancing types.
-    The default is "aux_loss".
-    """
+    moe_router_load_balancing_type: str = "aux_loss"
+    """The load balancing strategy for the router. "aux_loss" corresponds to the load balancing loss
+    used in GShard and SwitchTransformer; "seq_aux_loss" corresponds to the load balancing loss used
+    in DeepSeekV2 and DeepSeekV3, which computes the loss for each individual sample; "sinkhorn"
+    corresponds to the balancing algorithm used in S-BASE, and "none" implies no load balancing.
+    The default is "aux_loss"."""
 
     moe_router_topk: int = 2
     """Number of experts to route to for each token."""
@@ -526,10 +483,8 @@ class TransformerConfig(ModelParallelConfig):
     """Use legacy GroupedMLP rather than TEGroupedMLP.
     Note: The legacy one will be deprecated soon."""
 
-    moe_aux_loss_coeff: Union[float, List[float]] = 0.0
-    """Scaling coefficient for the aux loss. A starting value of 1e-2 is recommended.
-    If a list of load balancing types is provided for `moe_router_load_balancing_type`,
-    a corresponding list of coefficients should be provided here."""
+    moe_aux_loss_coeff: float = 0  # 1e-2 would be a good start value for load balance loss.
+    """Scaling coefficient for the aux loss. A starting value of 1e-2 is recommended."""
 
     moe_z_loss_coeff: Optional[float] = None  # 1e-3 would be a good start value for z-loss
     """Scaling coefficient for the z-loss. A starting value of 1e-3 is recommended."""
@@ -573,9 +528,6 @@ class TransformerConfig(ModelParallelConfig):
     moe_permute_fusion: bool = False
     """Fuse token rearrangement ops during token dispatching."""
 
-    moe_router_fusion: bool = False
-    """Fuse ops in routing and aux loss calculation."""
-
     moe_apply_probs_on_input: bool = False
     """Apply probs on input of experts instead of applying after activation and glu."""
 
@@ -602,9 +554,7 @@ class TransformerConfig(ModelParallelConfig):
     # Cuda Graphs
     ##################
     enable_cuda_graph: bool = False
-    """When set to true, either partial CUDA graph (1/many CUDA graph per layer) or full iteration
-    CUDA graph (1 CUDA graph for whole iteration excluding optimizer) is enabled. --cuda-graph-scope
-    determines the scope of graph capture."""
+    """When set to true, TransformerLayer layers are swapped with a CUDA graphed version."""
 
     cuda_graph_use_single_mempool: bool = False
     """When set to true, cudagraphs will be captured inside a single mempool, in which all
@@ -625,11 +575,9 @@ class TransformerConfig(ModelParallelConfig):
     """When set to true, TransformerLayer layers are swapped with user provided CUDA graphs."""
 
     cuda_graph_scope: str = "full"
-    """Determines the CUDA graphs capturing scope. When external_cuda_graph is set to true,
-    valid values are "full" and "attn". "Full" scope captures a whole Transformer
-    layer. "Attn" scope only captures operations in TransformerLayer._forward_attention().
-    When enable_cuda_graph is set to true, "full_iteration" can be specified as cuda_graph_scope
-    to enable whole iteration CUDA graph. All other values enable layerwise CUDA graph."""
+    """When external_cuda_graph is set to true, cuda_graph_scope determines the CUDA graphs
+    capturing scope. Valid values are "full" and "attn". "Full" scope captures a whole Transformer
+    layer. "Attn" scope only captures operations in TransformerLayer._forward_attention()."""
 
     ####################
     # miscellaneous
@@ -644,21 +592,21 @@ class TransformerConfig(ModelParallelConfig):
     config_logger_dir: str = ""
     """When non-empty, dumps entry-point configs to config_logger_dir"""
 
+    qk_layernorm_hidden_dim: bool = False
+    """Whether to apply LayerNorm to the query and key embeddings on the hidden dimension rather than head dimension."""
+
+    use_partial_reduce_for_shared_embedding: bool = False
+    """Whether to use partional reduce for shared embedding."""
+
     flash_decode: bool = False
     """ Use the optimized flash decoding kernel during inference. """
 
-    use_te_activation_func: bool = False
-    """Whether to use ffn activation functions implemented by TransformerEngine"""
-
     use_te_rng_tracker: bool = False
     """ Whether to use the TE or MCore version of the RNG tracker. """
 
     inference_rng_tracker: bool = False
     """ Whether we should instantiate a separate RNG tracker for inference. """
 
-    inference_sampling_seed: int = 42
-    """ Random seed to use for sampling during inference. """
-
     symmetric_ar_type: Optional[str] = None
     """Type of symmetric all reduce to use"""
 
@@ -701,9 +649,25 @@ class TransformerConfig(ModelParallelConfig):
     quant_recipe: Optional[RecipeConfig] = None
     """Configuration of any quantization to be applied to the model"""
 
-    transformer_impl: str = "transformer_engine"
-    """Transformer implementation to use.
-    Options are 'transformer_engine' for Transformer Engine and 'local' for MCore."""
+    ####################
+    # PEFT
+    ####################
+    peft_type: str = None
+    """Type for finetuning"""
+    lora_target_modules: Optional[List[str]] = None
+    """Lora target modules"""
+    lora_dim: Optional[int] = None
+    """Lora rank."""
+    lora_alpha: Optional[int] = None
+    """Lora scale alpha."""
+    lora_dropout: Optional[float] = None
+    """Lora dropout prob"""
+    lora_dropout_position: Optional[str] = None
+    """Lora dropout pos"""
+    lora_in_init_method: Optional[str] = None
+    """Lora a init method"""
+    lora_out_init_method: Optional[str] = None
+    """Lora b init method"""
 
     def __post_init__(self):
         """Python dataclass method that is used to modify attributes after initialization.
@@ -775,13 +739,6 @@ class TransformerConfig(ModelParallelConfig):
         if self.fp8_param and not self.fp8:
             raise ValueError("fp8_param must be used together with fp8 mode.")
 
-        # FP4 validation
-        if self.fp4_param and not self.fp4:
-            raise ValueError("fp4_param must be used together with fp4 mode.")
-
-        if self.fp4 and self.fp8:
-            raise ValueError("fp4 and fp8 cannot be used simultaneously. Please choose one.")
-
         if self.apply_query_key_layer_scaling:
             self.attention_softmax_in_fp32 = True
 
@@ -824,38 +781,12 @@ class TransformerConfig(ModelParallelConfig):
                     f"moe_shared_expert_overlap only works with alltoall token dispatcher."
                 )
 
-        if isinstance(self.moe_router_load_balancing_type, list):
-            assert isinstance(self.moe_aux_loss_coeff, list) and len(
-                self.moe_aux_loss_coeff
-            ) == len(self.moe_router_load_balancing_type), (
-                "moe_aux_loss_coeff must be a list of the same length as "
-                "moe_router_load_balancing_type"
-            )
-
         if self.moe_expert_capacity_factor is not None:
             if self.moe_expert_capacity_factor < 0:
                 self.moe_expert_capacity_factor = None
-            if isinstance(self.moe_router_load_balancing_type, list):
-                for load_balancing_type in self.moe_router_load_balancing_type:
-                    if load_balancing_type not in [
-                        "aux_loss",
-                        "seq_aux_loss",
-                        "global_aux_loss",
-                        "none",
-                    ]:
-                        raise ValueError(
-                            "moe_expert_capacity_factor only works with aux_loss, "
-                            "seq_aux_loss, global_aux_loss or none load balancing"
-                        )
-            elif self.moe_router_load_balancing_type not in [
-                "aux_loss",
-                "seq_aux_loss",
-                "global_aux_loss",
-                "none",
-            ]:
+            if self.moe_router_load_balancing_type not in ["aux_loss", "seq_aux_loss", "none"]:
                 raise ValueError(
-                    "moe_expert_capacity_factor only works with aux_loss, "
-                    "seq_aux_loss, global_aux_loss or none load balancing"
+                    "moe_expert_capacity_factor only works with aux_loss or none load balancing"
                 )
 
         if self.moe_pad_expert_input_to_capacity:
@@ -925,15 +856,7 @@ class TransformerConfig(ModelParallelConfig):
 
         if self.recompute_granularity == "selective":
             if len(self.recompute_modules) > 0:
-                allowed_modules = {
-                    "core_attn",
-                    "moe_act",
-                    "layernorm",
-                    "mla_up_proj",
-                    "mlp",
-                    "moe",
-                    "shared_experts",
-                }
+                allowed_modules = {"core_attn", "moe_act", "layernorm", "mla_up_proj", "mlp", "moe"}
                 invalid_modules = set(self.recompute_modules) - allowed_modules
                 assert not invalid_modules, (
                     f"Invalid choices for recompute_modules: {invalid_modules}. "
@@ -959,28 +882,9 @@ class TransformerConfig(ModelParallelConfig):
                     "Please check that the core_attn recompute is really needed."
                 )
 
-            if "shared_experts" in self.recompute_modules:
-                if (
-                    self.moe_shared_expert_intermediate_size is not None
-                    and self.moe_shared_expert_overlap
-                ):
-                    raise ValueError(
-                        "shared_experts recompute cannot work with --moe-shared-expert-overlap."
-                    )
-
             if self.fp8:
                 if "moe_act" in self.recompute_modules or "layernorm" in self.recompute_modules:
-                    if self.fp8_recipe == 'delayed':
-                        raise ValueError(
-                            "Delayed scaling does not support moe_act and layernorm recompute "
-                            "for fp8."
-                        )
-                    if not is_te_min_version("2.6.0dev0"):
-                        raise ValueError(
-                            "moe_act and layernorm recompute for fp8 needs "
-                            "transformer-engine>=2.6.0dev0, "
-                            f"but your version is {get_te_version()}."
-                        )
+                    raise ValueError("moe_act and layernorm recompute cannot work with fp8.")
 
         if self.moe_layer_recompute:
             warnings.warn(
@@ -1056,9 +960,7 @@ class TransformerConfig(ModelParallelConfig):
                 self.virtual_pipeline_model_parallel_size = detected_vpp_size
 
             # Check whether the layout is valid.
-            self.pipeline_model_parallel_layout.validate_layer_layout(
-                num_layers=self.num_layers, mtp_num_layers=self.mtp_num_layers
-            )
+            self.pipeline_model_parallel_layout.validate_layer_layout(num_layers=self.num_layers)
 
         # Uneven PP
         elif (
@@ -1180,10 +1082,10 @@ class TransformerConfig(ModelParallelConfig):
             self.attention_softmax_in_fp32 = True
 
         if self.bias_activation_fusion:
-            if self.activation_func not in [F.gelu, F.silu, quick_gelu]:
+            if self.activation_func not in [F.gelu, F.silu]:
                 raise ValueError(
                     "When bias_activation_fusion is True, activation function should be either "
-                    "gelu, swiglu, or quick_geglu"
+                    "gelu or swiglu"
                 )
             if (
                 self.activation_func == F.gelu
@@ -1191,34 +1093,9 @@ class TransformerConfig(ModelParallelConfig):
                 and not self.add_bias_linear
             ):
                 raise ValueError(
-                    "When bias_activation_fusion is True, gated_linear_unit is False "
+                    "When bias_activation_fusion is True, gated_linear_unit is False, "
                     "and activation function is gelu, add_bias_linear must also be True."
                 )
-            if self.activation_func == quick_gelu and not self.gated_linear_unit:
-                raise ValueError(
-                    "When bias_activation_fusion is True and activation function is quick_gelu, "
-                    "gated_linear_unit must be True."
-                )
-            if self.glu_linear_offset != 0.0 and self.activation_func != quick_gelu:
-                raise ValueError(
-                    "When bias_activation_fusion is True and glu_linear_offset is non-zero, "
-                    "activation function must be quick_gelu."
-                )
-
-            if self.use_te_activation_func:
-                raise ValueError(
-                    "bias_activation_fusion and use_te_activation_func cannot be both true. "
-                    "If you use bias in MLP FC1, we recommend setting bias_activation_fusion "
-                    "to True and use_te_activation_func to False."
-                )
-
-        if self.use_te_activation_func:
-            if self.activation_func not in (F.gelu, F.silu, F.relu):
-                raise ValueError(
-                    "TransformerEngine only support gelu, geglu, silu, swiglu, relu, reglu. "
-                    "If you don't want to use TransformerEngine activation function, set "
-                    "use_te_activation_func to False"
-                )
 
         if self.activation_func_fp8_input_store:
             if self.activation_func != F.silu or not self.gated_linear_unit:
@@ -1232,10 +1109,10 @@ class TransformerConfig(ModelParallelConfig):
                 )
             else:
                 if self.rotary_interleaved:
-                    if not is_te_min_version("2.3.0"):
+                    if not is_te_min_version("2.3.0.dev0"):
                         raise ValueError(
                             "rotary_interleaved does not work with apply_rope_fusion for "
-                            "TE < 2.3.0. Please install TE >= 2.3.0"
+                            "TE < 2.3.0.dev0. Please install TE >= 2.3.0.dev0"
                         )
 
                 from megatron.core.models.common.embeddings.rope_utils import (
@@ -1245,29 +1122,12 @@ class TransformerConfig(ModelParallelConfig):
 
                 if fused_apply_rotary_pos_emb is None and fused_apply_rotary_pos_emb_thd is None:
                     raise ValueError(
-                        "apply_rope_fusion is not available. Please install TE >= 1.4."
+                        "apply_rope_fusion is not available. Please install TE >= 1.4 or Apex."
                     )
 
         if self.multi_latent_attention and self.rotary_interleaved:
             raise ValueError("rotary_interleaved does not work with multi_latent_attention.")
 
-        # Set the embedding init method
-        if self.embedding_init_method_std is None:
-            # By default, use the same init std as you use for every other non-output layer.
-            self.embedding_init_method_std = self.init_method_std
-
-        if self.embedding_init_method is None:
-            if self.init_method is None or (self.embedding_init_method_std != self.init_method_std):
-                # In this case, we set both the init method and the embedding init method to
-                #  whatever std value requested (or defaulted) for the embedding_init_layer
-                self.embedding_init_method = init_method_normal(self.embedding_init_method_std)
-            else:
-                # Replicate the current behavior where if you are not changing the std of the
-                #  embedding init differently and the init method is set, we fallback to the
-                #  init method for this layer. Since we are here after an OR we know that
-                #  init_method is not None
-                self.embedding_init_method = self.init_method
-
         if self.init_method is None:
             self.init_method = init_method_normal(self.init_method_std)
 
@@ -1379,56 +1239,6 @@ class TransformerConfig(ModelParallelConfig):
             ):
                 raise ValueError("fused permutation is not available. Please install TE >= 2.1.0.")
 
-        if self.overlap_moe_expert_parallel_comm:
-            # TODO: remove this after we fix the hang issue with torch version < 2.6.0
-            assert is_torch_min_version(
-                "2.6.0"
-            ), "A2A Overlap encounters hang issue with torch version < 2.6.0"
-            if self.pipeline_model_parallel_size > 1:
-                assert self.virtual_pipeline_model_parallel_size is not None, (
-                    "If enabling EP A2A overlap, virtual_pipeline_model_parallel_size "
-                    "must be specified when pipeline_model_parallel_size > 1"
-                )
-            # Expert model parallelism requirements
-            assert (
-                self.expert_model_parallel_size > 1
-            ), 'overlap_moe_expert_parallel_comm is only supported with expert model parallelism'
-            assert self.moe_token_dispatcher_type in [
-                'alltoall',
-                'flex',
-            ], 'overlap_moe_expert_parallel_comm is supported with alltoall/flex token dispatcher'
-
-            assert (
-                self.recompute_granularity != 'full'
-            ), 'disable full recomputation when enabling overlap_moe_expert_parallel_comm'
-            assert (
-                self.recompute_method is None
-            ), 'disable recomputation method when enabling overlap_moe_expert_parallel_comm'
-            assert (
-                self.recompute_num_layers is None
-            ), 'recompute_num_layers must be None when enabling overlap_moe_expert_parallel_comm'
-
-            # Check if bf16 or fp16 is used
-            assert (
-                self.bf16 or self.fp16
-            ), 'overlap_moe_expert_parallel_comm is only supported with bf16 or fp16 model'
-
-            assert (
-                not self.moe_shared_expert_overlap
-            ), 'disable moe_shared_expert_overlap when enabling overlap_moe_expert_parallel_comm'
-            assert (
-                self.mtp_num_layers is None or self.mtp_num_layers == 1
-            ), 'MTP layernum only supports 1 when enabling overlap_moe_expert_parallel_comm.'
-
-        # Check delay_wgrad_compute compatibility
-        if self.delay_wgrad_compute:
-            assert (
-                self.overlap_moe_expert_parallel_comm
-            ), 'overlap_moe_expert_parallel_comm must be enabled when enabling delay_wgrad_compute'
-            assert (
-                not self.moe_use_legacy_grouped_gemm
-            ), 'delay_wgrad_compute is not supported with legacy groupedgemm implementation'
-
         if self.context_parallel_size > 1 and self.cp_comm_type is not None:
             if isinstance(self.cp_comm_type, list):
                 assert len(self.cp_comm_type) == self.num_layers, (
@@ -1481,6 +1291,9 @@ class TransformerConfig(ModelParallelConfig):
                     f"the number of layers ({self.num_layers})"
                 )
 
+        if self.moe_fb_overlap:
+            self.delay_wgrad_compute = True
+
 
 @dataclass
 class MLATransformerConfig(TransformerConfig):
@@ -1535,17 +1348,12 @@ class MLATransformerConfig(TransformerConfig):
     beta_slow: float = 1
     """Beta slow for YaRN RoPE, used by yarn."""
 
-    mscale: float = 1.0
+    mscale: float = 0.707
     """Mscale for YaRN RoPE in Multi-Latent Attention, used by yarn."""
 
-    mscale_all_dim: float = 0.0
+    mscale_all_dim: float = 0.707
     """Mscale all dimensions for YaRN RoPE in Multi-Latent Attention, used by yarn."""
 
-    cache_mla_latents: bool = False
-    """Cache the low dimensional tensors for MLA rather than full KV cache.
-       This is only for the dynamic inference backend and requires that 
-       Flash MLA is installed."""
-
     def __post_init__(self):
         super().__post_init__()
         if self.multi_latent_attention and self.apply_rope_fusion and self.rope_type != "yarn":
@@ -1564,8 +1372,3 @@ class MLATransformerConfig(TransformerConfig):
                 "Assigned original_max_position_embeddings to max_position_embeddings if not set,"
                 "and assigned max_position_embeddings back to the original value."
             )
-
-        if self.cache_mla_latents:
-            assert (
-                self.apply_rope_fusion is False
-            ), "Rope Fusion is not compatible with caching latents"

