diff --git a/megatron/post_training/arguments.py b/megatron/post_training/arguments.py
index 1e3bce937..f9fdbbb02 100644
--- a/megatron/post_training/arguments.py
+++ b/megatron/post_training/arguments.py
@@ -81,6 +81,47 @@ def add_modelopt_args(parser):
         help="Checkpoint format of teacher model, if different from student's.",
     )
 
+    # Speculative decoding
+    group.add_argument(
+        '--export-num-medusa-heads',
+        type=int,
+        default=0,
+        help='Number of Medusa heads for speculative decoding.',
+    )
+    group.add_argument(
+        '--export-eagle-algorithm',
+        type=str,
+        choices=['eagle1', 'eagle3', 'eagle-mtp'],
+        default="eagle-mtp",
+        help='Chosing the between different flavors of EAGLE algorithms.',
+    )
+    group.add_argument(
+        '--export-num-eagle-layers',
+        type=int,
+        default=0,
+        help='Number of EAGLE layers for speculative decoding.',
+    )
+    group.add_argument(
+        '--export-draft-vocab-size',
+        type=int,
+        default=0,
+        help='The reduced vocabulary size of the draft model.',
+    )
+    group.add_argument(
+        '--export-num-mtp',
+        type=int,
+        default=0,
+        help='Number of MTP modules for speculative decoding.',
+    )
+    group.add_argument(
+        '--export-freeze-mtp',
+        type=int,
+        nargs="*",
+        default=[],
+        help='Index of MTP that will be frozen in training.',
+    )
+
+
 
     # Finetuning
     group.add_argument(
@@ -102,13 +143,6 @@ def add_modelopt_args(parser):
         help='Use Llama-4 expert scaling on input instead of output.',
     )
 
-    # Speculative decoding
-    group.add_argument(
-        '--export-offline-model',
-        action="store_true",
-        help='If set, the base model will have no decoder layer. Only the embedding layer and output layer are initialized.',
-    )
-
     return parser
 
 

