diff --git a/tests/unit_tests/transformer/moe/test_sequential_mlp.py b/tests/unit_tests/transformer/moe/test_sequential_mlp.py
index a80e2a2ff..132cce57f 100644
--- a/tests/unit_tests/transformer/moe/test_sequential_mlp.py
+++ b/tests/unit_tests/transformer/moe/test_sequential_mlp.py
@@ -11,7 +11,7 @@ from megatron.core.tensor_parallel.random import model_parallel_cuda_manual_seed
 from megatron.core.transformer.mlp import MLPSubmodules
 from megatron.core.transformer.moe.experts import SequentialMLP
 from megatron.core.transformer.moe.moe_layer import MoELayer
-from megatron.core.transformer.moe.moe_utils import get_default_pg_collection
+from megatron.core.transformer.moe.moe_utils import get_default_model_comm_pgs
 from megatron.core.transformer.transformer_config import TransformerConfig
 from megatron.core.utils import is_te_min_version
 from tests.unit_tests.test_utilities import Utils
@@ -75,7 +75,7 @@ class TestTEParallelSequentialMLP:
         Utils.initialize_model_parallel(tensor_model_parallel_size=2, expert_model_parallel_size=2)
         model_parallel_cuda_manual_seed(123)
         num_moe_experts = 4
-        pg_collection = get_default_pg_collection()
+        model_comm_pgs = get_default_model_comm_pgs()
         self.transformer_config = TransformerConfig(
             num_layers=2,
             hidden_size=12,
@@ -108,7 +108,7 @@ class TestTEParallelSequentialMLP:
             self.num_local_experts,
             self.transformer_config,
             self.local_mlp_spec,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )
 
         model_parallel_cuda_manual_seed(123)
@@ -116,7 +116,7 @@ class TestTEParallelSequentialMLP:
             self.num_local_experts,
             self.transformer_config,
             self.te_mlp_spec,
-            pg_collection=pg_collection,
+            model_comm_pgs=model_comm_pgs,
         )
 
     @pytest.mark.internal
@@ -168,13 +168,13 @@ class TestTEParallelSequentialMLP:
     @pytest.mark.internal
     def test_gpu_forward_with_one_local_expert(self):
         model_parallel_cuda_manual_seed(123)
-        pg_collection = get_default_pg_collection()
+        model_comm_pgs = get_default_model_comm_pgs()
         local_sequential_mlp = SequentialMLP(
-            1, self.transformer_config, self.local_mlp_spec, pg_collection=pg_collection
+            1, self.transformer_config, self.local_mlp_spec, model_comm_pgs=model_comm_pgs
         )
         model_parallel_cuda_manual_seed(123)
         te_sequential_mlp = SequentialMLP(
-            1, self.transformer_config, self.te_mlp_spec, pg_collection=pg_collection
+            1, self.transformer_config, self.te_mlp_spec, model_comm_pgs=model_comm_pgs
         )
         seq_len = 4
         batch_size = 2

