diff --git a/megatron/post_training/algos/distillation.py b/megatron/post_training/algos/distillation.py
index 604d1044e..f47dfccad 100644
--- a/megatron/post_training/algos/distillation.py
+++ b/megatron/post_training/algos/distillation.py
@@ -536,6 +536,7 @@ def get_tensor_shapes_adjust_fn_for_distillation(
         rank = get_pipeline_model_parallel_rank()
         teacher_config = get_model_config(model.teacher_model)
         teacher_model_type = get_model_type(model.teacher_model)
+        teacher_encoder_decoder_xattn = get_model_xattn(model.teacher_model)
 
         teacher_recv_tensor_shapes = get_tensor_shapes(
             rank=rank - 1,
@@ -544,6 +545,7 @@ def get_tensor_shapes_adjust_fn_for_distillation(
             micro_batch_size=micro_batch_size,
             decoder_seq_length=decoder_seq_length,
             config=teacher_config,
+            encoder_decoder_xattn=teacher_encoder_decoder_xattn,
         )
         teacher_send_tensor_shapes = get_tensor_shapes(
             rank=rank,
@@ -552,6 +554,7 @@ def get_tensor_shapes_adjust_fn_for_distillation(
             micro_batch_size=micro_batch_size,
             decoder_seq_length=decoder_seq_length,
             config=teacher_config,
+            encoder_decoder_xattn=teacher_encoder_decoder_xattn,
         )
         model.set_student_input_tensor_shape(recv_tensor_shapes)
 

