diff --git a/tests/unit_tests/dist_checkpointing/test_pipeline_parallel_layout.py b/tests/unit_tests/dist_checkpointing/test_pipeline_parallel_layout.py
index 7a5f7d2dd..5674ec366 100644
--- a/tests/unit_tests/dist_checkpointing/test_pipeline_parallel_layout.py
+++ b/tests/unit_tests/dist_checkpointing/test_pipeline_parallel_layout.py
@@ -34,7 +34,7 @@ def initialize_gpt_model(
     vocab_size=128,
     virtual_pipeline_model_parallel_size=None,
     is_moe=False,
-    **config_kwargs,
+    **config_kwargs
 ):
     torch.manual_seed(seed)
     model_parallel_cuda_manual_seed(seed)
@@ -149,10 +149,6 @@ def create_args():
     args.no_load_optim = True
     args.no_load_rng = True
     args.use_distributed_optimizer = True
-    args.use_megatron_fsdp = False
-    args.dist_ckpt_save_pre_mcore_014 = False
-    args.dist_ckpt_optim_fully_reshardable = False
-    args.distrib_optim_fully_reshardable_mem_efficient = False
 
     yield args
 
@@ -326,10 +322,11 @@ def test_save_and_load_checkpoint_vpp(
     opt_param_scheduler = None
     num_floating_point_operations_so_far = 456
 
-    with (
-        TempNamedDir(tmp_path_dist_ckpt / 'test_gpt_model_reconfiguration_model_A') as ckpt_dir_A,
-        TempNamedDir(tmp_path_dist_ckpt / 'test_gpt_model_reconfiguration_model_B') as ckpt_dir_B,
-    ):
+    with TempNamedDir(
+        tmp_path_dist_ckpt / 'test_gpt_model_reconfiguration_model_A'
+    ) as ckpt_dir_A, TempNamedDir(
+        tmp_path_dist_ckpt / 'test_gpt_model_reconfiguration_model_B'
+    ) as ckpt_dir_B:
         set_ckpt_path(ckpt_dir_A)
         save_checkpoint(
             iteration, model, optimizer, opt_param_scheduler, num_floating_point_operations_so_far

