diff --git a/tests/unit_tests/dist_checkpointing/test_local.py b/tests/unit_tests/dist_checkpointing/test_local.py
index 1b8597e1f..bb86bf6dd 100644
--- a/tests/unit_tests/dist_checkpointing/test_local.py
+++ b/tests/unit_tests/dist_checkpointing/test_local.py
@@ -142,7 +142,6 @@ class TestLocalCheckpointing:
             # ShardedObjects and ShardedTensors should be replaced
             assert issubclass(i[-1], ShardedBase)
 
-    @pytest.mark.internal
     @pytest.mark.parametrize(('tp,pp'), [(2, 4), (1, 1)])
     @pytest.mark.parametrize(('use_ramdisk'), [True, False])
     @pytest.mark.parametrize(('async_save'), [True, False])
@@ -160,18 +159,14 @@ class TestLocalCheckpointing:
         )  # FIXME: fails with additional arguments (e.g.,'weight_decay')
         if use_ramdisk:
             tmp_path_dist_ckpt = Path("/dev/shm")
-
-        original_empty = torch.empty
-
-        def deterministic_empty(*args, **kwargs):
-            return original_empty(*args, **kwargs).zero_()
-
-        with (
-            TempNamedDir(tmp_path_dist_ckpt / "test_local", sync=True) as local_ckpt_dir,
-            mock.patch('megatron.training.checkpointing.get_args', new=lambda: mock_args),
-            mock.patch('megatron.training.async_utils.get_args', new=lambda: mock_args),
-            mock.patch("megatron.training.checkpointing.update_num_microbatches"),
-            mock.patch('torch.empty', new=deterministic_empty),
+        with TempNamedDir(
+            tmp_path_dist_ckpt / "test_local", sync=True
+        ) as local_ckpt_dir, mock.patch(
+            'megatron.training.checkpointing.get_args', new=lambda: mock_args
+        ), mock.patch(
+            'megatron.training.async_utils.get_args', new=lambda: mock_args
+        ), mock.patch(
+            "megatron.training.checkpointing.update_num_microbatches"
         ):
             local_ckpt_dir = local_ckpt_dir / "subdir"  # Test handling of non-existent directories
             init_basic_mock_args(mock_args, tp, pp)
@@ -179,7 +174,6 @@ class TestLocalCheckpointing:
             mock_args.non_persistent_ckpt_type = 'local'
             mock_args.non_persistent_local_ckpt_algo = algo
             mock_args.async_save = async_save
-            mock_args.ckpt_fully_parallel_save = True  # ensure proper sharding_type is set
             checkpointing_context = {
                 'local_checkpoint_manager': LocalCheckpointManager(local_ckpt_dir)
             }
@@ -250,7 +244,6 @@ class TestLocalCheckpointing:
 
         Utils.destroy_model_parallel()
 
-    @pytest.mark.internal
     @pytest.mark.parametrize(('tp,pp'), [(1, 1), (2, 4)])
     @pytest.mark.parametrize(('use_ramdisk'), [True, False])
     @pytest.mark.parametrize(('async_save'), [True, False])
@@ -267,13 +260,16 @@ class TestLocalCheckpointing:
             tmp_path_dist_ckpt = Path("/dev/shm")
 
         def test_save_wrapper(save_wrapper, subdir):
-            with (
-                TempNamedDir(tmp_path_dist_ckpt / subdir, sync=True) as local_ckpt_dir,
-                mock.patch('megatron.training.checkpointing.get_args', new=lambda: mock_args),
-                mock.patch('megatron.training.async_utils.get_args', new=lambda: mock_args),
-                mock.patch("megatron.training.checkpointing.update_num_microbatches"),
-                mock.patch.object(LocalCheckpointManager, '_save', new=save_wrapper),
-                caplog.at_level(logging.INFO),
+            with TempNamedDir(tmp_path_dist_ckpt / subdir, sync=True) as local_ckpt_dir, mock.patch(
+                'megatron.training.checkpointing.get_args', new=lambda: mock_args
+            ), mock.patch(
+                'megatron.training.async_utils.get_args', new=lambda: mock_args
+            ), mock.patch(
+                "megatron.training.checkpointing.update_num_microbatches"
+            ), mock.patch.object(
+                LocalCheckpointManager, '_save', new=save_wrapper
+            ), caplog.at_level(
+                logging.INFO
             ):
 
                 local_ckpt_dir = (
@@ -284,7 +280,6 @@ class TestLocalCheckpointing:
                 mock_args.non_persistent_ckpt_type = 'local'
                 mock_args.non_persistent_local_ckpt_algo = algo
                 mock_args.async_save = async_save
-                mock_args.ckpt_fully_parallel_save = True  # ensure proper sharding_type is set
                 checkpointing_context = {
                     'local_checkpoint_manager': LocalCheckpointManager(local_ckpt_dir)
                 }

