diff --git a/tests/unit_tests/dist_checkpointing/test_global_metadata_reuse.py b/tests/unit_tests/dist_checkpointing/test_global_metadata_reuse.py
index 49f6567e8..80a10c921 100644
--- a/tests/unit_tests/dist_checkpointing/test_global_metadata_reuse.py
+++ b/tests/unit_tests/dist_checkpointing/test_global_metadata_reuse.py
@@ -31,12 +31,12 @@ class TestGlobalMetadataReuse:
         opt_param_scheduler = None
 
         mock_args = parse_args(ignore_unknown_args=True)
-        with (
-            TempNamedDir(
-                tmp_path_dist_ckpt / "test_global_metadata_reuse"
-            ) as non_persistent_ckpt_dir,
-            mock.patch('megatron.training.checkpointing.get_args', new=lambda: mock_args),
-            mock.patch("megatron.training.checkpointing.update_num_microbatches"),
+        with TempNamedDir(
+            tmp_path_dist_ckpt / "test_global_metadata_reuse"
+        ) as non_persistent_ckpt_dir, mock.patch(
+            'megatron.training.checkpointing.get_args', new=lambda: mock_args
+        ), mock.patch(
+            "megatron.training.checkpointing.update_num_microbatches"
         ):
             init_basic_mock_args(mock_args, tp, pp)
             init_checkpointing_mock_args(mock_args, non_persistent_ckpt_dir)
@@ -101,21 +101,18 @@ class TestGlobalMetadataReuse:
         opt_param_scheduler = None
 
         mock_args = parse_args(ignore_unknown_args=True)
-        with (
-            TempNamedDir(
-                tmp_path_dist_ckpt / "test_global_metadata_reuse"
-            ) as non_persistent_ckpt_dir,
-            mock.patch('megatron.training.checkpointing.get_args', new=lambda: mock_args),
-            mock.patch("megatron.training.checkpointing.update_num_microbatches"),
+        with TempNamedDir(
+            tmp_path_dist_ckpt / "test_global_metadata_reuse"
+        ) as non_persistent_ckpt_dir, mock.patch(
+            'megatron.training.checkpointing.get_args', new=lambda: mock_args
+        ), mock.patch(
+            "megatron.training.checkpointing.update_num_microbatches"
         ):
             init_basic_mock_args(mock_args, tp, pp)
             init_checkpointing_mock_args(mock_args, non_persistent_ckpt_dir)
             mock_args.non_persistent_ckpt_type = "global"
             mock_args.ckpt_assume_constant_structure = True
             mock_args.ckpt_fully_parallel_save = True
-            mock_args.use_distributed_optimizer = True
-            # We need full reshardability as we change TP/PP
-            mock_args.dist_ckpt_optim_fully_reshardable = True
 
             save_ckpt_context = {}
 
@@ -140,7 +137,6 @@ class TestGlobalMetadataReuse:
             Utils.initialize_model_parallel(pp, tp)
             model, optimizer = setup_model_and_optimizer(1, pp, tp)
             init_basic_mock_args(mock_args, pp, tp)
-            mock_args.use_distributed_optimizer = True
             mock_args.no_load_rng = True
 
             resume_ckpt_context = {}

