diff --git a/megatron/legacy/data/data_samplers.py b/megatron/legacy/data/data_samplers.py
index 1bf1bf5ee..78c7e1af4 100644
--- a/megatron/legacy/data/data_samplers.py
+++ b/megatron/legacy/data/data_samplers.py
@@ -9,7 +9,6 @@ import numpy as np
 from torch.utils.data import Dataset
 from megatron.training import get_args
 from megatron.core import mpu
-from megatron.core.datasets.utils import Split
 
 
 def build_pretraining_data_loader(dataset, consumed_samples):
@@ -18,23 +17,9 @@ def build_pretraining_data_loader(dataset, consumed_samples):
     if dataset is None:
         return None
     args = get_args()
-    
-    if hasattr(dataset,'split'):
-        split = dataset.split
-    elif hasattr(dataset,'index_split'):
-        split = dataset.index_split
-    else:
-        split = None
 
-    if split == Split.valid and args.full_validation:
-        batch_sampler = MegatronPretrainingSampler(
-            total_samples=len(dataset),
-            consumed_samples=0,
-            micro_batch_size=args.micro_batch_size,
-            data_parallel_rank=mpu.get_data_parallel_rank(),
-            data_parallel_size=mpu.get_data_parallel_world_size())
-    elif args.dataloader_type == 'single':
-        # Megatron sampler
+    # Megatron sampler
+    if args.dataloader_type == 'single':
         batch_sampler = MegatronPretrainingSampler(
             total_samples=len(dataset),
             consumed_samples=consumed_samples,

