diff --git a/tools/checkpoint/loader_legacy.py b/tools/checkpoint/loader_legacy.py
index 0aefadacd..2771105f0 100644
--- a/tools/checkpoint/loader_legacy.py
+++ b/tools/checkpoint/loader_legacy.py
@@ -5,8 +5,6 @@ import os
 import sys
 import types
 
-from functools import partial
-
 import torch
 
 from tools.checkpoint.utils import _ConverterFakeProcessGroup
@@ -118,9 +116,7 @@ def _load_checkpoint(queue, args):
 
     # Determine how to make our models
     if args.model_type == 'GPT':
-        from model_provider import model_provider as common_model_provider
-        from gpt_builders import gpt_builder
-        model_provider = partial(common_model_provider, gpt_builder)
+        from pretrain_gpt import model_provider
         margs.model_type = ModelType.encoder_or_decoder
     elif args.model_type == 'BERT':
         from pretrain_bert import model_provider

