diff --git a/tests/unit_tests/run_ci_test.sh b/tests/unit_tests/run_ci_test.sh
old mode 100755
new mode 100644
index 0864b607b..2c93a0642
--- a/tests/unit_tests/run_ci_test.sh
+++ b/tests/unit_tests/run_ci_test.sh
@@ -160,7 +160,7 @@ export ONE_LOGGER_JOB_CATEGORY=test
 
 for i in $(seq $UNIT_TEST_REPEAT); do
     echo "Running prod test suite."
-    CMD=$(echo uv run python -m torch.distributed.run ${DISTRIBUTED_ARGS[@]} \
+    CMD=$(echo python -m torch.distributed.run ${DISTRIBUTED_ARGS[@]} \
         -m pytest \
         -xvs \
         --cov-report=term \
@@ -171,10 +171,9 @@ for i in $(seq $UNIT_TEST_REPEAT); do
     eval "$CMD"
 
     if [[ "$TAG" == "latest" ]]; then
-        CMD=$(echo uv run python -m torch.distributed.run ${DISTRIBUTED_ARGS[@]} -m pytest \
+        CMD=$(echo python -m torch.distributed.run ${DISTRIBUTED_ARGS[@]} -m pytest \
             -xvs \
             --experimental \
-             ${IGNORE_ARGS[@]} \
             -m "'experimental and ${MARKER_ARG}'" $BUCKET)
 
         eval "$CMD"

