diff --git a/tests/unit_tests/test_utils.py b/tests/unit_tests/test_utils.py
index 64a25e600..98dcf0030 100644
--- a/tests/unit_tests/test_utils.py
+++ b/tests/unit_tests/test_utils.py
@@ -61,9 +61,8 @@ def test_experimental_cls_static():
 
 
 def test_experimental_cls_exception_init():
-    with (
-        patch.object(config, 'ENABLE_EXPERIMENTAL', False),
-        pytest.raises(util.ExperimentalNotEnabledError),
+    with patch.object(config, 'ENABLE_EXPERIMENTAL', False), pytest.raises(
+        util.ExperimentalNotEnabledError
     ):
         a = A()
         assert a.some_method() == success_string
@@ -71,9 +70,8 @@ def test_experimental_cls_exception_init():
 
 
 def test_experimental_cls_exception_static():
-    with (
-        patch.object(config, 'ENABLE_EXPERIMENTAL', False),
-        pytest.raises(util.ExperimentalNotEnabledError),
+    with patch.object(config, 'ENABLE_EXPERIMENTAL', False), pytest.raises(
+        util.ExperimentalNotEnabledError
     ):
         assert A.some_static_method() == success_string
 
@@ -232,7 +230,6 @@ def test_cross_check_param_hashes_across_dp_replicas():
 
 @pytest.mark.parametrize("use_distributed_optimizer", [False, True])
 @pytest.mark.flaky_in_dev
-@pytest.mark.internal
 def test_param_norm(use_distributed_optimizer: bool):
     world = int(os.getenv('WORLD_SIZE', '1'))
     rank = int(os.getenv('RANK', '0'))
@@ -369,14 +366,14 @@ def test_straggler_detector():
 
     # Check if the instance is in disabled state.
     straggler_detector_disabled()
-    # Enable it now, must call report.
-    straggler_detector_enable()
-    # Check if all ranks have straggler detector enabled.
-    straggler_detector_enabled()
-    # Time some operation.
-    straggler_detector_timeit()
-    # Report only from rank 0.
-    straggler_detector_report()
+    # # Enable it now, must call report.
+    # straggler_detector_enable()
+    # # Check if all ranks have straggler detector enabled.
+    # straggler_detector_enabled()
+    # # Time some operation.
+    # straggler_detector_timeit()
+    # # Report only from rank 0.
+    # straggler_detector_report()
     # Check that exception is not suppressed.
     straggler_detector_exception_propagate()
     util.StragglerDetector._configured = False

