diff --git a/megatron/core/package_info.py b/megatron/core/package_info.py
index d13e2d9aa..c0df4fd58 100644
--- a/megatron/core/package_info.py
+++ b/megatron/core/package_info.py
@@ -2,9 +2,9 @@
 
 
 MAJOR = 0
-MINOR = 15
+MINOR = 14
 PATCH = 0
-PRE_RELEASE = 'rc5'
+PRE_RELEASE = 'rc1'
 
 # Use the following formatting: (major, minor, patch, pre-release)
 VERSION = (MAJOR, MINOR, PATCH, PRE_RELEASE)
@@ -15,8 +15,10 @@ __version__ = '.'.join(map(str, VERSION[:3])) + ''.join(VERSION[3:])
 __package_name__ = 'megatron_core'
 __contact_names__ = 'NVIDIA'
 __contact_emails__ = 'nemo-toolkit@nvidia.com'  # use NeMo Email
-__homepage__ = 'https://docs.nvidia.com/megatron-core/developer-guide/latest/user-guide/index.html'
-__repository_url__ = 'https://github.com/NVIDIA/Megatron-LM/tree/main/megatron/core'
+__homepage__ = (
+    'https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/'  # use NeMo homepage
+)
+__repository_url__ = 'https://github.com/NVIDIA/Megatron-LM/megatron/core'
 __download_url__ = 'https://github.com/NVIDIA/Megatron-LM/releases'
 __description__ = (
     'Megatron Core - a library for efficient and scalable training of transformer based models'

