diff --git a/megatron/core/packed_seq_params.py b/megatron/core/packed_seq_params.py
index 330d0e034..dff0cc599 100644
--- a/megatron/core/packed_seq_params.py
+++ b/megatron/core/packed_seq_params.py
@@ -16,5 +16,5 @@ class PackedSeqParams:
     cu_seqlens_kv: Tensor = None
     cu_seqlens_q_padded: Tensor = None
     cu_seqlens_kv_padded: Tensor = None
-    max_seqlen_q: int = None
-    max_seqlen_kv: int = None
+    max_seqlen_q: Tensor = None
+    max_seqlen_kv: Tensor = None

