diff --git a/megatron/core/utils.py b/megatron/core/utils.py
index 0300f0f02..a75877a5b 100644
--- a/megatron/core/utils.py
+++ b/megatron/core/utils.py
@@ -3,7 +3,6 @@
 """Utility functions used throughout Megatron core"""
 
 import array
-import asyncio
 import functools
 import hashlib
 import inspect
@@ -402,9 +401,6 @@ def deprecate_inference_params(inference_context, inference_params):
 def get_tensor_model_parallel_group_if_none(tp_group, is_expert=False, check_initialized=True):
     """Issue a deprecation warning if tp_group is None and return the default tp group."""
     # TODO(zijiey): remove this function later.
-    if not torch.distributed.is_initialized():
-        return None
-
     if tp_group is None:
         if torch.distributed.is_initialized() and torch.distributed.get_rank() == 0:
             warnings.warn(
@@ -664,13 +660,7 @@ def log_single_rank(logger: logging.Logger, *args: Any, rank: int = 0, **kwargs:
         logger.log(*args, **kwargs)
 
 
-def log_on_each_pipeline_stage(
-    logger: logging.Logger,
-    *args: Any,
-    tp_group: Optional[torch.distributed.ProcessGroup] = None,
-    dp_cp_group: Optional[torch.distributed.ProcessGroup] = None,
-    **kwargs: Any,
-):
+def log_on_each_pipeline_stage(logger: logging.Logger, *args: Any, **kwargs: Any):
     """Log on first rank in each pipeline stage
 
     Args:
@@ -682,16 +672,10 @@ def log_on_each_pipeline_stage(
     """
     assert torch.distributed.is_initialized()
 
-    if tp_group is None and dp_cp_group is None:
-        tp_rank = parallel_state.get_tensor_model_parallel_rank()
-        dp_cp_rank = parallel_state.get_data_parallel_rank(with_context_parallel=True)
-    elif tp_group is not None and dp_cp_group is not None:
-        tp_rank = tp_group.rank()
-        dp_cp_rank = dp_cp_group.rank()
-    else:
-        raise ValueError("tp_group and dp_cp_group must be provided or not provided together")
-
-    if tp_rank == 0 and dp_cp_rank == 0:
+    if (
+        parallel_state.get_data_parallel_rank(with_context_parallel=True) == 0
+        and parallel_state.get_tensor_model_parallel_rank() == 0
+    ):
         logger.log(*args, **kwargs)
 
 
@@ -810,6 +794,28 @@ def make_tp_sharded_tensor_for_checkpoint(
     if replica_id is None:
         replica_id = (0, 0, dp_replica_id)
 
+    if hasattr(tensor, "fully_shard_param_local_shard"):
+        assert len(replica_id) == 3, f"Expected replica_id format (PP, TP, DP), got: {replica_id}"
+        replica_id = (*replica_id[:2], tensor.fsdp_instance_id)
+
+        sh_ten = ShardedTensor.from_rank_offsets_flat(
+            key,
+            tensor.fully_shard_param_local_shard,
+            tensor.shape,
+            *prepend_offsets,
+            (
+                tp_axis + prepend_axis_num,
+                parallel_state.get_tensor_model_parallel_rank(),
+                parallel_state.get_tensor_model_parallel_world_size(),
+            ),
+            flattened_range=slice(*tensor.fully_shard_param_local_index),
+            replica_id=replica_id,
+            prepend_axis_num=prepend_axis_num,
+            **kwargs,
+        )
+        setattr(sh_ten, "is_data_parallel_fully_shard", True)
+        return sh_ten
+
     return ShardedTensor.from_rank_offsets(
         key,
         tensor,
@@ -843,6 +849,23 @@ def make_sharded_tensor_for_checkpoint(tensor, key, prepend_offsets=(), replica_
     if replica_id is None:
         replica_id = (0, parallel_state.get_tensor_model_parallel_rank(), dp_replica_id)
 
+    if hasattr(tensor, "fully_shard_param_local_shard"):
+        assert len(replica_id) == 3, f"Expected replica_id format (PP, TP, DP), got: {replica_id}"
+        replica_id = (*replica_id[:2], tensor.fsdp_instance_id)
+
+        sh_ten = ShardedTensor.from_rank_offsets_flat(
+            key,
+            tensor.fully_shard_param_local_shard,
+            tensor.shape,
+            *prepend_offsets,
+            flattened_range=slice(*tensor.fully_shard_param_local_index),
+            replica_id=replica_id,
+            prepend_axis_num=prepend_axis_num,
+            **kwargs,
+        )
+        setattr(sh_ten, "is_data_parallel_fully_shard", True)
+        return sh_ten
+
     return ShardedTensor.from_rank_offsets(
         key,
         tensor,
@@ -916,9 +939,7 @@ except Exception:
     dist_all_gather_func = torch.distributed._all_gather_base
 
 
-def drain_embedding_wgrad_compute(
-    config, embedding_activation_buffer, grad_output_buffer, weight, tp_group
-):
+def drain_embedding_wgrad_compute(config, embedding_activation_buffer, grad_output_buffer, weight):
     """Helper for performing embedding wgrad GEMM's during the pipeline drain phase, pipelines the
     AllGather and GEMM's.
 
@@ -932,17 +953,23 @@ def drain_embedding_wgrad_compute(
 
     import fused_weight_gradient_mlp_cuda
 
-    from megatron.core.parallel_state import get_global_memory_buffer
+    from megatron.core.parallel_state import (
+        get_global_memory_buffer,
+        get_tensor_model_parallel_group,
+        get_tensor_model_parallel_world_size,
+    )
 
     input = embedding_activation_buffer.pop(0)
-    world_size = tp_group.size()
+    world_size = get_tensor_model_parallel_world_size()
     dim_size = list(input.size())
     dim_size[0] = dim_size[0] * world_size
 
     all_gathered_input = [None, None]
     if config.sequence_parallel:
         all_gather_buffer = get_global_memory_buffer().get_tensor(dim_size, input.dtype, "mpu_0")
-        handle = dist_all_gather_func(all_gather_buffer, input, group=tp_group, async_op=False)
+        handle = dist_all_gather_func(
+            all_gather_buffer, input, group=get_tensor_model_parallel_group(), async_op=False
+        )
 
         all_gathered_input[0] = all_gather_buffer
         all_gather_buffer = None
@@ -956,9 +983,6 @@ def drain_embedding_wgrad_compute(
             grad_output, all_gathered_input
         )
 
-        if hasattr(weight, "__fsdp_param__"):
-            weight.main_grad = weight.get_main_grad()
-
         if config.gradient_accumulation_fusion:
             if weight.main_grad.dtype == torch.float32:
                 fused_weight_gradient_mlp_cuda.wgrad_gemm_accum_fp32(
@@ -980,7 +1004,9 @@ def drain_embedding_wgrad_compute(
         if config.sequence_parallel:
             name = "mpu_" + str((i + 1) % 2)
             all_gather_buffer = get_global_memory_buffer().get_tensor(dim_size, input.dtype, name)
-            handle = dist_all_gather_func(all_gather_buffer, input, group=tp_group, async_op=True)
+            handle = dist_all_gather_func(
+                all_gather_buffer, input, group=get_tensor_model_parallel_group(), async_op=True
+            )
 
             all_gathered_input[(i + 1) % 2] = all_gather_buffer
             all_gather_buffer = None
@@ -1824,9 +1850,9 @@ def get_batch_on_this_cp_rank(batch: Dict[str, Any]):
                     val.shape[seq_dim] // (2 * cp_size),
                     *val.shape[(seq_dim + 1) :],
                 )
-                index = torch.zeros(2, dtype=torch.int64, device=val.device)
-                index[0].fill_(cp_rank)
-                index[1].fill_(2 * cp_size - cp_rank - 1)
+                index = torch.tensor(
+                    [cp_rank, (2 * cp_size - cp_rank - 1)], device="cpu", pin_memory=True
+                ).cuda(non_blocking=True)
                 val = val.index_select(seq_dim, index)
                 val = val.view(*val.shape[0:seq_dim], -1, *val.shape[(seq_dim + 2) :])
                 batch[key] = val
@@ -1961,39 +1987,3 @@ def nvtx_decorator(message: Optional[str] = None, color: Optional[str] = None):
         return func
 
     return decorator
-
-
-def unwrap_model(model, module_instances=None):
-    """Unwrap_model to return the final model instance"""
-    if module_instances is None:
-        from megatron.core.distributed import DistributedDataParallel as DDP
-        from megatron.core.distributed import TorchFullyShardedDataParallel as torch_FSDP
-        from megatron.core.distributed.fsdp.mcore_fsdp_adapter import (
-            FullyShardedDataParallel as megatron_FSDP,
-        )
-        from megatron.core.transformer.module import Float16Module
-
-        module_instances = (DDP, torch_FSDP, megatron_FSDP, Float16Module)
-
-    return_list = True
-    if not isinstance(model, list):
-        model = [model]
-        return_list = False
-    unwrapped_model = []
-    for model_module in model:
-        while isinstance(model_module, module_instances):
-            model_module = model_module.module
-        unwrapped_model.append(model_module)
-    if not return_list:
-        return unwrapped_model[0]
-    return unwrapped_model
-
-
-def get_asyncio_loop():
-    """Creates an asyncio loop if necessary and then returns the current asyncio loop."""
-    try:
-        loop = asyncio.get_running_loop()
-    except RuntimeError as e:
-        loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(loop)
-    return loop

