diff --git a/megatron/core/models/gpt/moe_module_specs.py b/megatron/core/models/gpt/moe_module_specs.py
index e1ea7c163..7a0fc1340 100755
--- a/megatron/core/models/gpt/moe_module_specs.py
+++ b/megatron/core/models/gpt/moe_module_specs.py
@@ -9,8 +9,6 @@ from megatron.core.transformer.moe.shared_experts import SharedExpertMLP
 from megatron.core.transformer.spec_utils import ModuleSpec
 
 try:
-    import transformer_engine as te  # pylint: disable=unused-import
-
     from megatron.core.extensions.transformer_engine_spec_provider import TESpecProvider
 
     HAVE_TE = True
@@ -42,25 +40,19 @@ def get_moe_module_spec_for_backend(
     num_experts: Optional[int] = None,
     moe_grouped_gemm: Optional[bool] = False,
     moe_use_legacy_grouped_gemm: Optional[bool] = False,
-    use_te_activation_func: bool = False,
 ) -> ModuleSpec:
     """Helper function to get module spec for MoE"""
     assert num_experts is not None
 
     linear_fc1 = backend.column_parallel_linear()
     linear_fc2 = backend.row_parallel_linear()
-    activation_func = backend.activation_func()
 
-    mlp = MLPSubmodules(
-        linear_fc1=linear_fc1, linear_fc2=linear_fc2, activation_func=activation_func
-    )
+    mlp = MLPSubmodules(linear_fc1=linear_fc1, linear_fc2=linear_fc2)
 
     expert_module, expert_submodule = backend.grouped_mlp_modules(
         moe_grouped_gemm is not None and moe_grouped_gemm,
         moe_use_legacy_grouped_gemm is not None and moe_use_legacy_grouped_gemm,
     )
-    if expert_submodule is not None:
-        expert_submodule.activation_func = activation_func
 
     experts = ModuleSpec(module=expert_module, submodules=expert_submodule)
 

