diff --git a/megatron/inference/text_generation/generation.py b/megatron/inference/text_generation/generation.py
index 31c4b82e0..1d033bb55 100644
--- a/megatron/inference/text_generation/generation.py
+++ b/megatron/inference/text_generation/generation.py
@@ -8,6 +8,7 @@ import torch.nn.functional as F
 from megatron.training import get_args, get_tokenizer
 from megatron.core import mpu
 from megatron.training.utils import get_ltor_masks_and_position_ids
+from megatron.core.transformer.cuda_graphs import create_cudagraphs
 from megatron.core.inference.contexts import StaticInferenceContext
 from .communication import (
     copy_from_last_to_first_pipeline_stage,
@@ -230,6 +231,9 @@ def generate_tokens_probs_and_return_on_first_stage(
             # logits will be meanigful only in the last pipeline stage.
             logits = forward_step(tokens2use, positions2use, attention_mask2use)
 
+            if args.enable_cuda_graph and not prefill:
+                create_cudagraphs()
+
             if mpu.is_pipeline_last_stage():
                 if prevent_newline_after_colon:
                     logits[tokens2use[:, -1] == tokenizer.tokenize(':')[0], -1, tokenizer.tokenize('\n')[0]] = -1e10 # disable "\n" after ":"

