diff --git a/megatron/core/models/retro/decoder_spec.py b/megatron/core/models/retro/decoder_spec.py
index 653934814..a706949dd 100644
--- a/megatron/core/models/retro/decoder_spec.py
+++ b/megatron/core/models/retro/decoder_spec.py
@@ -3,7 +3,6 @@
 """Specs for Retro decoder."""
 
 import typing
-from typing import Optional
 
 from megatron.core.models.gpt.gpt_layer_specs import (
     get_gpt_layer_local_spec,
@@ -41,8 +40,6 @@ except ImportError:
     HAVE_APEX = False
 
 try:
-    import transformer_engine as te  # pylint: disable=unused-import
-
     from megatron.core.extensions.transformer_engine import (
         TEColumnParallelLinear,
         TEDotProductAttention,
@@ -124,10 +121,7 @@ def get_retro_decoder_layer_local_spec(
 
 
 def get_retro_decoder_block_spec(
-    config: RetroConfig,
-    use_transformer_engine: bool,
-    vp_stage: Optional[int] = None,
-    pp_rank: Optional[int] = None,
+    config: RetroConfig, use_transformer_engine: bool
 ) -> TransformerBlockSubmodules:
     """Retro decoder block spec.
 
@@ -142,8 +136,6 @@ def get_retro_decoder_block_spec(
     Args:
         config (RetroConfig): Retro config.
         use_transformer_engine (bool): If True, use Transformer Engine (instead of local modules.
-        vp_stage (Optional[int]): Virtual pipeline stage number.
-        pp_rank (Optional[int]): Pipeline parallel rank.
 
     Returns:
         Transformer block submodules for the given spec.
@@ -158,7 +150,7 @@ def get_retro_decoder_block_spec(
     ), "retro does not currently support virtual pipeline parallelism."
 
     # Num layers.
-    num_layers = get_num_layers_to_build(config, vp_stage=vp_stage, pp_rank=pp_rank)
+    num_layers = get_num_layers_to_build(config)
 
     # Retro layer numbers.
     retro_layer_start = 6 if num_layers <= 15 else 9

