diff --git a/megatron/core/activations.py b/megatron/training/activations.py
similarity index 83%
rename from megatron/core/activations.py
rename to megatron/training/activations.py
index 8b422d73a..4d0fed14f 100644
--- a/megatron/core/activations.py
+++ b/megatron/training/activations.py
@@ -7,17 +7,13 @@ from megatron.core.jit import jit_fuser
 
 @jit_fuser
 def squared_relu(x: torch.Tensor) -> torch.Tensor:
-    """Squared ReLU activation"""
     return torch.pow(F.relu(x), 2)
 
 
 @jit_fuser
 def quick_gelu(x: torch.Tensor) -> torch.Tensor:
-    """Quick GELU activation"""
     return x * torch.sigmoid(1.702 * x)
 
-
 @jit_fuser
 def fast_gelu(x: torch.Tensor) -> torch.Tensor:
-    """Fast GELU activation"""
     return 0.5 * x * (1.0 + torch.tanh(x * 0.7978845608 * (1.0 + 0.044715 * x * x)))

