diff --git a/megatron/training/async_utils.py b/megatron/training/async_utils.py
index 6cf0c1e55..3a923dbaa 100644
--- a/megatron/training/async_utils.py
+++ b/megatron/training/async_utils.py
@@ -7,7 +7,6 @@ the async checkpoint save calls.
 import logging
 
 from megatron.core.dist_checkpointing.strategies.async_utils import AsyncCallsQueue, AsyncRequest
-from megatron.core.dist_checkpointing.strategies.filesystem_async import _results_queue
 from megatron.training import get_args
 from megatron.training.utils import print_rank_0
 
@@ -64,15 +63,3 @@ def is_empty_async_queue() -> bool:
         bool: True if there is any ongoing async call.
     """
     return _async_calls_queue.get_num_unfinalized_calls() == 0
-
-
-def reset_persistent_async_worker():
-    global _async_calls_queue, _results_queue
-    if _async_calls_queue is not None:
-        _async_calls_queue.close(abort=True)
-        del _async_calls_queue
-    if _results_queue is not None:
-        _results_queue._manager.shutdown()
-        del _results_queue
-    _results_queue = None
-    _async_calls_queue = None

