diff --git a/megatron/core/inference/headers.py b/megatron/core/inference/headers.py
deleted file mode 100644
index ff894cc19..000000000
--- a/megatron/core/inference/headers.py
+++ /dev/null
@@ -1,17 +0,0 @@
-# Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
-
-from enum import Enum
-
-
-class Headers(Enum):
-    """
-    Enum representing headers used for communication with the inference-coordinator.
-    """
-
-    CONNECT = 0
-    ACK = 1
-    SUBMIT_REQUEST = 2
-    ENGINE_REPLY = 3
-    PAUSE = 4
-    UNPAUSE = 5
-    STOP = 6

