diff --git a/megatron/core/inference/inference_request.py b/megatron/core/inference/inference_request.py
index 09623b11c..7111d11cb 100644
--- a/megatron/core/inference/inference_request.py
+++ b/megatron/core/inference/inference_request.py
@@ -1,10 +1,9 @@
-# Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
 
-import time
 import warnings
-from dataclasses import asdict, dataclass, field
-from enum import Enum, auto
-from typing import Any, Dict, List, Optional
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, List, Optional
 
 import torch
 
@@ -19,7 +18,6 @@ class Status(Enum):
     ACTIVE_AND_GENERATING_TOKENS = 2
     ACTIVE_BUT_NOT_GENERATING_TOKENS = 3
     COMPLETED = 4
-    FAILED = 5
 
 
 @dataclass(kw_only=True)
@@ -58,67 +56,6 @@ class InferenceRequest:
             )
             self.sampling_params = self.inference_parameters
 
-    def serializable(self):
-        """
-        Converts the instance into a serializable dictionary.
-        Returns:
-            dict: A dictionary representation of the instance suitable for serialization.
-        """
-
-        return asdict(self)
-
-
-class DynamicInferenceEventType(Enum):
-    """Dynamic inference event type."""
-
-    ADD = auto()
-    PAUSE = auto()
-    FINISH = auto()
-    FAIL = auto()
-    ERROR_TRANSIENT = auto()
-    ERROR_NONTRANSIENT = auto()
-
-
-@dataclass(kw_only=True)
-class DynamicInferenceEvent:
-    """A lifecycle event for a dynamic inference requests.
-
-    An event is currently one of the following:
-
-    - request added
-    - request paused
-    - request finished
-    - request failed
-    - request error (transient)
-    - request error (non-transient, i.e. fatal)
-    """
-
-    timestamp: Optional[float] = None
-    type: DynamicInferenceEventType
-    payload: Optional[Any] = None
-
-    def __post_init__(self):
-
-        # Timestamp.
-        assert self.timestamp is None, "timestamp automatically set."
-        self.timestamp = time.time()
-
-        # Validate type.
-        assert isinstance(self.type, DynamicInferenceEventType)
-
-        # Validate payload.
-        if self.type in (
-            DynamicInferenceEventType.ERROR_TRANSIENT,
-            DynamicInferenceEventType.ERROR_NONTRANSIENT,
-        ):
-            assert self.payload is not None
-        else:
-            assert self.payload is None
-
-    def __str__(self):
-        payload_str = "" if self.payload is None else f", {type(self.payload).__name__}"
-        return f"[{self.timestamp:.3f}] {self.type.name}{payload_str}"
-
 
 @dataclass(kw_only=True)
 class DynamicInferenceRequest(InferenceRequest):
@@ -132,55 +69,6 @@ class DynamicInferenceRequest(InferenceRequest):
     generated_tokens: List[int] = field(default_factory=list)
     prompt: Optional[str] = None
     prompt_tokens: Optional[torch.Tensor] = None
-    latency: Optional[float] = None
-    events: List[DynamicInferenceEvent] = field(default_factory=list)
-
-    def __str__(self):
-        return ", ".join(
-            (
-                f"id {self.request_id}",
-                f"{self.status.name}" if self.status is not None else "[NOT ADDED]",
-                f"prompt len {len(self.prompt_tokens)}",
-                f"gen len {len(self.generated_tokens)}",
-                f"num events {len(self.events)}",
-            )
-        )
-
-    def add_event(self, type: DynamicInferenceEventType, payload: Optional[Any] = None) -> None:
-        """Add event."""
-        self.events.append(DynamicInferenceEvent(type=type, payload=payload))
-
-    def add_event_add(self):
-        """Add 'add' event."""
-        return self.add_event(DynamicInferenceEventType.ADD)
-
-    def add_event_pause(self):
-        """Add 'pause' event."""
-        return self.add_event(DynamicInferenceEventType.PAUSE)
-
-    def add_event_finish(self):
-        """Add 'finish' event."""
-        return self.add_event(DynamicInferenceEventType.FINISH)
-
-    def add_event_fail(self):
-        """Add 'fail' event."""
-        return self.add_event(DynamicInferenceEventType.FAIL)
-
-    def add_event_error_transient(self, error: Exception):
-        """Add transient error event."""
-        return self.add_event(DynamicInferenceEventType.ERROR_TRANSIENT, error)
-
-    def add_event_error_nontransient(self, error: Exception):
-        """Add non-transient error event."""
-        return self.add_event(DynamicInferenceEventType.ERROR_NONTRANSIENT, error)
-
-    def succeeded(self) -> bool:
-        """Request experienced no non-transient errors."""
-        return self.status == Status.COMPLETED
-
-    def failed(self) -> bool:
-        """Request experienced non-transient error."""
-        return self.status == Status.FAILED
 
 
 @dataclass(kw_only=True)

